<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><title>MetaDraft Reaction Report</title></head><body> <h2 id="page-top">MetaDraft Reaction Report</h2> <h3>Analysis</h3> <p> <strong>Input fasta:</strong> test.fasta<br/> <strong>Metaproteome used:</strong> bigg2-eco41<br/> <strong>Report date:</strong> 20-11-02<br/> <strong>MetaDraft version:</strong> 0.9.3 </p> <!--<h3>Notes</h3> <p></p>--> <h3>Reactions (56)</h3> <h4>Reactions selected</h4> <table width="80%" border="0" cellspacing="1" cellpadding="1">   <tbody><tr><td>Reaction</td><td>Name</td><td>Model</td><td>Source</td></tr><tr><td><a href="#bigg2-eco41_R_ACGAMT">R_ACGAMT</a></td><td>UDP-N-acetylglucosamine:undecaprenylphosphate N-acetylglucosamine -1-phosphate transferase</td><td>bigg2-eco41</td><td>stu0163</td></tr><tr><td><a href="#bigg2-eco41_R_ACGApts">R_ACGApts</a></td><td>N-Acetyl-D-glucosamine transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_ACMANApts">R_ACMANApts</a></td><td>N-acetyl-D-mannosamine transport via PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_ACONT">R_ACONT</a></td><td>Aconitate hydratase</td><td>bigg2-eco41</td><td>stu1268</td></tr><tr><td><a href="#bigg2-eco41_R_AGDC">R_AGDC</a></td><td>N-acetylglucosamine-6-phosphate deacetylase</td><td>bigg2-eco41</td><td>stu0501</td></tr><tr><td><a href="#bigg2-eco41_R_ALAALAr">R_ALAALAr</a></td><td>D-alanine-D-alanine ligase (reversible)</td><td>bigg2-eco41</td><td>stu1583</td></tr><tr><td><a href="#bigg2-eco41_R_ALCD19">R_ALCD19</a></td><td>Alcohol dehydrogenase (glycerol)</td><td>bigg2-eco41</td><td>stu0936</td></tr><tr><td><a href="#bigg2-eco41_R_AMALT1">R_AMALT1</a></td><td>Amylomaltase (maltotriose)</td><td>bigg2-eco41</td><td>stu1013</td></tr><tr><td><a href="#bigg2-eco41_R_AMALT2">R_AMALT2</a></td><td>Amylomaltase (maltotetraose)</td><td>bigg2-eco41</td><td>stu1013</td></tr><tr><td><a href="#bigg2-eco41_R_AMALT3">R_AMALT3</a></td><td>Amylomaltase (maltopentaose)</td><td>bigg2-eco41</td><td>stu1013</td></tr><tr><td><a href="#bigg2-eco41_R_AMALT4">R_AMALT4</a></td><td>Amylomaltase (maltohexaose)</td><td>bigg2-eco41</td><td>stu1013</td></tr><tr><td><a href="#bigg2-eco41_R_ASPabc">R_ASPabc</a></td><td>L-aspartate transport via ABC system</td><td>bigg2-eco41</td><td>stu1164</td></tr><tr><td><a href="#bigg2-eco41_R_CYSS">R_CYSS</a></td><td>Cysteine synthase</td><td>bigg2-eco41</td><td>stu0846</td></tr><tr><td><a href="#bigg2-eco41_R_DHAPT">R_DHAPT</a></td><td>Dihydroxyacetone phosphotransferase</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_DHBSr">R_DHBSr</a></td><td>2,3-dihydroxybenzoate adenylate synthase</td><td>bigg2-eco41</td><td>stu0761</td></tr><tr><td><a href="#bigg2-eco41_R_FRUpts">R_FRUpts</a></td><td>D-fructose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_FRUpts2">R_FRUpts2</a></td><td>Fructose transport via PEP:Pyr PTS (f6p generating)</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_GALTpts">R_GALTpts</a></td><td>Galactitol transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_GAMpts">R_GAMpts</a></td><td>D-glucosamine transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_GLCP">R_GLCP</a></td><td>Glycogen phosphorylase</td><td>bigg2-eco41</td><td>stu1012</td></tr><tr><td><a href="#bigg2-eco41_R_GLCpts">R_GLCpts</a></td><td>D-glucose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_GLNabc">R_GLNabc</a></td><td>L-glutamine transport via ABC system</td><td>bigg2-eco41</td><td>stu0606,stu1492</td></tr><tr><td><a href="#bigg2-eco41_R_GLU5K">R_GLU5K</a></td><td>Glutamate 5-kinase</td><td>bigg2-eco41</td><td>stu1711</td></tr><tr><td><a href="#bigg2-eco41_R_GLUabc">R_GLUabc</a></td><td>L-glutamate transport via ABC system</td><td>bigg2-eco41</td><td>stu1164</td></tr><tr><td><a href="#bigg2-eco41_R_ICDHyr">R_ICDHyr</a></td><td>Isocitrate dehydrogenase (NADP)</td><td>bigg2-eco41</td><td>stu1266</td></tr><tr><td><a href="#bigg2-eco41_R_ILETA">R_ILETA</a></td><td>Isoleucine transaminase</td><td>bigg2-eco41</td><td>stu0590</td></tr><tr><td><a href="#bigg2-eco41_R_Kt2r">R_Kt2r</a></td><td>Potassium reversible transport  via proton symport</td><td>bigg2-eco41</td><td>stu0264</td></tr><tr><td><a href="#bigg2-eco41_R_LCADi_copy1">R_LCADi_copy1</a></td><td></td><td>bigg2-eco41</td><td>stu0936</td></tr><tr><td><a href="#bigg2-eco41_R_LEUTAi">R_LEUTAi</a></td><td>Leucine transaminase (irreversible)</td><td>bigg2-eco41</td><td>stu0590</td></tr><tr><td><a href="#bigg2-eco41_R_MALTpts">R_MALTpts</a></td><td>Maltose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_MANpts">R_MANpts</a></td><td>D-mannose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_MCITS">R_MCITS</a></td><td>2-methylcitrate synthase</td><td>bigg2-eco41</td><td>stu1267</td></tr><tr><td><a href="#bigg2-eco41_R_METS_1">R_METS_1</a></td><td>Methionine synthase</td><td>bigg2-eco41</td><td>stu0785</td></tr><tr><td><a href="#bigg2-eco41_R_MICITDr">R_MICITDr</a></td><td>2-methylisocitrate dehydratase</td><td>bigg2-eco41</td><td>stu1268</td></tr><tr><td><a href="#bigg2-eco41_R_MNLpts">R_MNLpts</a></td><td>Mannitol transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_OBTFL">R_OBTFL</a></td><td>2-Oxobutanoate formate lyase</td><td>bigg2-eco41</td><td>stu1640</td></tr><tr><td><a href="#bigg2-eco41_R_OCBT">R_OCBT</a></td><td>Ornithine carbamoyltransferase</td><td>bigg2-eco41</td><td>stu0603</td></tr><tr><td><a href="#bigg2-eco41_R_PFL">R_PFL</a></td><td>Pyruvate formate lyase</td><td>bigg2-eco41</td><td>stu1640</td></tr><tr><td><a href="#bigg2-eco41_R_PGM">R_PGM</a></td><td>Phosphoglycerate mutase</td><td>bigg2-eco41</td><td>stu0697</td></tr><tr><td><a href="#bigg2-eco41_R_PHETA1">R_PHETA1</a></td><td>Phenylalanine transaminase</td><td>bigg2-eco41</td><td>stu0590</td></tr><tr><td><a href="#bigg2-eco41_R_PRPPS">R_PRPPS</a></td><td>Phosphoribosylpyrophosphate synthetase</td><td>bigg2-eco41</td><td>stu1460</td></tr><tr><td><a href="#bigg2-eco41_R_PTPATi">R_PTPATi</a></td><td>Pantetheine-phosphate adenylyltransferase</td><td>bigg2-eco41</td><td>stu1648</td></tr><tr><td><a href="#bigg2-eco41_R_SBTpts">R_SBTpts</a></td><td>D-sorbitol transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_SERAT">R_SERAT</a></td><td>Serine O-acetyltransferase</td><td>bigg2-eco41</td><td>stu0848</td></tr><tr><td><a href="#bigg2-eco41_R_SHSL1">R_SHSL1</a></td><td>O-succinylhomoserine lyase (L-cysteine)</td><td>bigg2-eco41</td><td>stu0847</td></tr><tr><td><a href="#bigg2-eco41_R_SPODM">R_SPODM</a></td><td>Superoxide dismutase</td><td>bigg2-eco41</td><td>stu0720</td></tr><tr><td><a href="#bigg2-eco41_R_SUCpts">R_SUCpts</a></td><td>Sucrose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_THZPSN">R_THZPSN</a></td><td>Thiazole phosphate synthesis</td><td>bigg2-eco41</td><td>stu1461</td></tr><tr><td><a href="#bigg2-eco41_R_TREpts">R_TREpts</a></td><td>Trehalose transport via PEP:Pyr PTS</td><td>bigg2-eco41</td><td>stu1264,stu1265</td></tr><tr><td><a href="#bigg2-eco41_R_TRPS1">R_TRPS1</a></td><td>Tryptophan synthase (indoleglycerol phosphate)</td><td>bigg2-eco41</td><td>stu1588</td></tr><tr><td><a href="#bigg2-eco41_R_TRPS2">R_TRPS2</a></td><td>Tryptophan synthase (indole)</td><td>bigg2-eco41</td><td>stu1588</td></tr><tr><td><a href="#bigg2-eco41_R_TRPS3">R_TRPS3</a></td><td>Tryptophan synthase (indoleglycerol phosphate)</td><td>bigg2-eco41</td><td>stu1588</td></tr><tr><td><a href="#bigg2-eco41_R_UAGCVT">R_UAGCVT</a></td><td>UDP-N-acetylglucosamine 1-carboxyvinyltransferase</td><td>bigg2-eco41</td><td>stu1167</td></tr><tr><td><a href="#bigg2-eco41_R_UDPG4E">R_UDPG4E</a></td><td>UDPglucose 4-epimerase</td><td>bigg2-eco41</td><td>stu1169</td></tr><tr><td><a href="#bigg2-eco41_R_UPPRT">R_UPPRT</a></td><td>Uracil phosphoribosyltransferase</td><td>bigg2-eco41</td><td>stu0355</td></tr><tr><td><a href="#bigg2-eco41_R_VALTA">R_VALTA</a></td><td>Valine transaminase</td><td>bigg2-eco41</td><td>stu0590</td></tr></tbody> </table> <h4>Reactions not selected</h4> <table width="80%" border="0" cellspacing="1" cellpadding="1">   <tbody><tr><td>Reaction</td><td>Name</td><td>Model</td><td>Source</td></tr></tbody></table> <h3>Annotation</h3> <hr/><table id="bigg2-eco41_R_ACGAMT"><caption>R_ACGAMT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ACGAMT</strong><br/>(irreversble)</td><td><strong>UDP-N-acetylglucosamine:undecaprenylphosphate N-acetylglucosamine -1-phosphate transferase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Central intermediary metabolism: Sugar-nucleotide biosynthesis, conversions 1.6.4 metabolism; macromolecules (cellular constituent) biosynthesis; enterobacterial common antigen (surface glycolipid) 6.3 cell structure; surface antigens (ECA, O antigen of LPS)<br/><strong>locus_tag: </strong>b3784<br/><strong>raw_location: </strong>3965939..3967042<br/><strong>GO_function: </strong>GO:0042280 - cell surface antigen activity, host-interacting<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3776; JW3758; wecA<br/><strong>EC_number: </strong>2.4.1.-<br/><strong>product: </strong>UDP-GlcNAc:undecaprenylphosphate GlcNAc-1-phosphate transferase<br/><strong>GO_process: </strong>GO:0009246 - enterobacterial common antigen biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>rfe<br/><strong>protein_id: </strong>NP_418231.1<br/>" style="color: green;">b3784</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0163)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_udcpp_c + M_uacgam_c > M_ump_c + M_unaga_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Undecaprenyl phosphate + UDP-N-acetyl-D-glucosamine > UMP C9H11N2O9P + Undecaprenyl diphospho N-acetyl-glucosamine </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_udcpp_c</strong></td><td bgcolor="#CCFFCC"><strong>Undecaprenyl phosphate</strong></td></tr><tr><td>charge: -2</td><td>C55H89O4P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_uacgam_c</strong></td><td bgcolor="#CCFFCC"><strong>UDP-N-acetyl-D-glucosamine</strong></td></tr><tr><td>charge: -2</td><td>C17H25N3O17P2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ump_c</strong></td><td bgcolor="#FFCCCC"><strong>UMP C9H11N2O9P</strong></td></tr><tr><td>charge: -2</td><td>C9H11N2O9P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_unaga_c</strong></td><td bgcolor="#FFCCCC"><strong>Undecaprenyl diphospho N-acetyl-glucosamine</strong></td></tr><tr><td>charge: -2</td><td>C63H103NO12P2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ACGAMT">http://identifiers.org/bigg.reaction/ACGAMT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95251">http://identifiers.org/metanetx.reaction/MNXR95251</a></td></tr></table><hr/><table id="bigg2-eco41_R_ACGApts"><caption>R_ACGApts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ACGApts</strong><br/>(irreversble)</td><td><strong>N-Acetyl-D-glucosamine transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>(((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b0679) or (<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1101)))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_acgam_e + M_pep_c > M_pyr_c + M_acgam6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>N-Acetyl-D-glucosamine + Phosphoenolpyruvate > Pyruvate + N-Acetyl-D-glucosamine 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_acgam_e</strong></td><td bgcolor="#CCFFCC"><strong>N-Acetyl-D-glucosamine</strong></td></tr><tr><td>charge: 0</td><td>C8H15NO6</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_acgam6p_c</strong></td><td bgcolor="#FFCCCC"><strong>N-Acetyl-D-glucosamine 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C8H14NO9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ACGApts">http://identifiers.org/bigg.reaction/ACGApts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95253">http://identifiers.org/metanetx.reaction/MNXR95253</a></td></tr></table><hr/><table id="bigg2-eco41_R_ACMANApts"><caption>R_ACMANApts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ACMANApts</strong><br/>(irreversble)</td><td><strong>N-acetyl-D-mannosamine transport via PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1819 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1817 and b1818))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_acmana_e > M_pyr_c + M_acmanap_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + N-Acetyl-D-mannosamine > Pyruvate + N-Acetyl-D-mannosamine 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_acmana_e</strong></td><td bgcolor="#CCFFCC"><strong>N-Acetyl-D-mannosamine</strong></td></tr><tr><td>charge: 0</td><td>C8H15NO6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_acmanap_c</strong></td><td bgcolor="#FFCCCC"><strong>N-Acetyl-D-mannosamine 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C8H14NO9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ACMANApts">http://identifiers.org/bigg.reaction/ACMANApts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95275">http://identifiers.org/metanetx.reaction/MNXR95275</a></td></tr></table><hr/><table id="bigg2-eco41_R_ACONT"><caption>R_ACONT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ACONT</strong><br/>(reversible)</td><td><strong>Aconitate hydratase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: TCA cycle<br/><strong>locus_tag: </strong>b1276<br/><strong>raw_location: </strong>1333855..1336530<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>acn; ECK1271; JW1268<br/><strong>EC_number: </strong>4.2.1.3<br/><strong>note: </strong>aconitate hydrase 1<br/><strong>GO_process: </strong>GO:0009061 - anaerobic respiration; GO:0006099 - tricarboxylic acid cycle<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>acnA<br/><strong>product: </strong>aconitate hydratase 1<br/><strong>protein_id: </strong>NP_415792.1<br/>" style="color: green;">b1276</span> or b0118))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1268)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_cit_c = M_icit_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Citrate = Isocitrate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_cit_c</strong></td><td bgcolor="#CCFFCC"><strong>Citrate</strong></td></tr><tr><td>charge: -3</td><td>C6H5O7</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_icit_c</strong></td><td bgcolor="#FFCCCC"><strong>Isocitrate</strong></td></tr><tr><td>charge: -3</td><td>C6H5O7</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ACONT">http://identifiers.org/bigg.reaction/ACONT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.3">http://identifiers.org/ec-code/4.2.1.3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01324">http://identifiers.org/kegg.reaction/R01324</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95384">http://identifiers.org/metanetx.reaction/MNXR95384</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10336">http://identifiers.org/rhea/10336</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10337">http://identifiers.org/rhea/10337</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10338">http://identifiers.org/rhea/10338</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10339">http://identifiers.org/rhea/10339</a></td></tr></table><hr/><table id="bigg2-eco41_R_AGDC"><caption>R_AGDC (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_AGDC</strong><br/>(irreversble)</td><td><strong>N-acetylglucosamine-6-phosphate deacetylase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Central intermediary metabolism: Amino sugars<br/><strong>locus_tag: </strong>b0677<br/><strong>raw_location: </strong>complement(700826..701974)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0665; JW0663<br/><strong>EC_number: </strong>3.5.1.25<br/><strong>product: </strong>N-acetylglucosamine-6-phosphate deacetylase<br/><strong>GO_process: </strong>GO:0046349 - amino sugar biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>nagA<br/><strong>protein_id: </strong>NP_415203.1<br/>" style="color: green;">b0677</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0501)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_h2o_c + M_acgam6p_c > M_ac_c + M_gam6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>H2O H2O + N-Acetyl-D-glucosamine 6-phosphate > Acetate + D-Glucosamine 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_acgam6p_c</strong></td><td bgcolor="#CCFFCC"><strong>N-Acetyl-D-glucosamine 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C8H14NO9P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ac_c</strong></td><td bgcolor="#FFCCCC"><strong>Acetate</strong></td></tr><tr><td>charge: -1</td><td>C2H3O2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_gam6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucosamine 6-phosphate</strong></td></tr><tr><td>charge: -1</td><td>C6H13NO8P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/AGDC">http://identifiers.org/bigg.reaction/AGDC</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:NAG6PDEACET-RXN">http://identifiers.org/biocyc/META:NAG6PDEACET-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/3.5.1.25">http://identifiers.org/ec-code/3.5.1.25</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R02059">http://identifiers.org/kegg.reaction/R02059</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95528">http://identifiers.org/metanetx.reaction/MNXR95528</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22936">http://identifiers.org/rhea/22936</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22937">http://identifiers.org/rhea/22937</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22938">http://identifiers.org/rhea/22938</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22939">http://identifiers.org/rhea/22939</a></td></tr></table><hr/><table id="bigg2-eco41_R_ALAALAr"><caption>R_ALAALAr (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ALAALAr</strong><br/>(reversible)</td><td><strong>D-alanine-D-alanine ligase (reversible)</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Murein sacculus, peptidoglycan<br/><strong>locus_tag: </strong>b0381<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>complement(399053..400147)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0376; JW0372<br/><strong>EC_number: </strong>6.3.2.4<br/><strong>product: </strong>D-alanine-D-alanine ligase A<br/><strong>GO_process: </strong>GO:0009252 - peptidoglycan biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>ddlA<br/><strong>protein_id: </strong>NP_414915.1<br/>" style="color: green;">b0381</span> or b0092))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1583)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_atp_c + (-2.0) M_ala__D_c = M_pi_c + M_alaala_c + M_adp_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>ATP C10H12N5O13P3 + (-2.0) D-Alanine = Phosphate + D-Alanyl-D-alanine + ADP C10H12N5O10P2 + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ala__D_c</strong></td><td bgcolor="#CCFFCC"><strong>D-Alanine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_alaala_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Alanyl-D-alanine</strong></td></tr><tr><td>charge: 0</td><td>C6H12N2O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_adp_c</strong></td><td bgcolor="#FFCCCC"><strong>ADP C10H12N5O10P2</strong></td></tr><tr><td>charge: -3</td><td>C10H12N5O10P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ALAALAr">http://identifiers.org/bigg.reaction/ALAALAr</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:DALADALALIG-RXN">http://identifiers.org/biocyc/META:DALADALALIG-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/6.3.2.4">http://identifiers.org/ec-code/6.3.2.4</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01150">http://identifiers.org/kegg.reaction/R01150</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95678">http://identifiers.org/metanetx.reaction/MNXR95678</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11224">http://identifiers.org/rhea/11224</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11225">http://identifiers.org/rhea/11225</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11226">http://identifiers.org/rhea/11226</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11227">http://identifiers.org/rhea/11227</a></td></tr></table><hr/><table id="bigg2-eco41_R_ALCD19"><caption>R_ALCD19 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ALCD19</strong><br/>(reversible)</td><td><strong>Alcohol dehydrogenase (glycerol)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: Fermentation<br/><strong>locus_tag: </strong>b0356<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(377686..378795)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>adhC; ECK0353; JW0347<br/><strong>EC_number: </strong>1.1.1.1 1.1.1.284<br/><strong>note: </strong>alcohol dehydrogenase class III; formaldehyde dehydrogenase, glutathione-dependent<br/><strong>GO_process: </strong>GO:0006113 - fermentation<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 1731906<br/><strong>protein_id: </strong>NP_414890.1<br/><strong>gene: </strong>frmA<br/><strong>product: </strong>alcohol dehydrogenase class III/glutathione-dependent formaldehyde dehydrogenase<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b0356</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0936)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_h_c + M_nadh_c + M_glyald_c = M_glyc_c + M_nad_c </td></tr><tr><td><strong>Equation Name</strong></td><td>H+ + Nicotinamide adenine dinucleotide - reduced + D-Glyceraldehyde = Glycerol + Nicotinamide adenine dinucleotide </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h_c</strong></td><td bgcolor="#CCFFCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_nadh_c</strong></td><td bgcolor="#CCFFCC"><strong>Nicotinamide adenine dinucleotide - reduced</strong></td></tr><tr><td>charge: -2</td><td>C21H27N7O14P2</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glyald_c</strong></td><td bgcolor="#CCFFCC"><strong>D-Glyceraldehyde</strong></td></tr><tr><td>charge: 0</td><td>C3H6O3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glyc_c</strong></td><td bgcolor="#FFCCCC"><strong>Glycerol</strong></td></tr><tr><td>charge: 0</td><td>C3H8O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_nad_c</strong></td><td bgcolor="#FFCCCC"><strong>Nicotinamide adenine dinucleotide</strong></td></tr><tr><td>charge: -1</td><td>C21H26N7O14P2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ALCD19">http://identifiers.org/bigg.reaction/ALCD19</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/1.1.1.21">http://identifiers.org/ec-code/1.1.1.21</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/1.1.1.72">http://identifiers.org/ec-code/1.1.1.72</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01036">http://identifiers.org/kegg.reaction/R01036</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95709">http://identifiers.org/metanetx.reaction/MNXR95709</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/30027">http://identifiers.org/rhea/30027</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/30028">http://identifiers.org/rhea/30028</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/30029">http://identifiers.org/rhea/30029</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/30030">http://identifiers.org/rhea/30030</a></td></tr></table><hr/><table id="bigg2-eco41_R_AMALT1"><caption>R_AMALT1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_AMALT1</strong><br/>(irreversble)</td><td><strong>Amylomaltase (maltotriose)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Degradation of polysaccharides<br/><strong>locus_tag: </strong>b3416<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3546008..3548092)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3403; JW3379; malA<br/><strong>EC_number: </strong>2.4.1.25<br/><strong>product: </strong>4-alpha-glucanotransferase (amylomaltase)<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000272 - polysaccharide catabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>malQ<br/><strong>protein_id: </strong>NP_417875.1<br/>" style="color: green;">b3416</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1013)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_malttr_c + M_malt_c > M_glc__D_c + M_maltttr_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Maltotriose C18H32O16 + Maltose C12H22O11 > D-Glucose + Maltotetraose </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malttr_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltotriose C18H32O16</strong></td></tr><tr><td>charge: 0</td><td>C18H32O16</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malt_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glc__D_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_maltttr_c</strong></td><td bgcolor="#FFCCCC"><strong>Maltotetraose</strong></td></tr><tr><td>charge: 0</td><td>C24H42O21</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/AMALT1">http://identifiers.org/bigg.reaction/AMALT1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:AMYLOMALT-RXN">http://identifiers.org/biocyc/META:AMYLOMALT-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.4.1.25">http://identifiers.org/ec-code/2.4.1.25</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95800">http://identifiers.org/metanetx.reaction/MNXR95800</a></td></tr></table><hr/><table id="bigg2-eco41_R_AMALT2"><caption>R_AMALT2 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_AMALT2</strong><br/>(irreversble)</td><td><strong>Amylomaltase (maltotetraose)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Degradation of polysaccharides<br/><strong>locus_tag: </strong>b3416<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3546008..3548092)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3403; JW3379; malA<br/><strong>EC_number: </strong>2.4.1.25<br/><strong>product: </strong>4-alpha-glucanotransferase (amylomaltase)<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000272 - polysaccharide catabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>malQ<br/><strong>protein_id: </strong>NP_417875.1<br/>" style="color: green;">b3416</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1013)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_malt_c + M_maltttr_c > M_maltpt_c + M_glc__D_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Maltose C12H22O11 + Maltotetraose > Maltopentaose + D-Glucose </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malt_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_maltttr_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltotetraose</strong></td></tr><tr><td>charge: 0</td><td>C24H42O21</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_maltpt_c</strong></td><td bgcolor="#FFCCCC"><strong>Maltopentaose</strong></td></tr><tr><td>charge: 0</td><td>C30H52O26</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glc__D_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/AMALT2">http://identifiers.org/bigg.reaction/AMALT2</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN-14260">http://identifiers.org/biocyc/META:RXN-14260</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.4.1.25">http://identifiers.org/ec-code/2.4.1.25</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95801">http://identifiers.org/metanetx.reaction/MNXR95801</a></td></tr></table><hr/><table id="bigg2-eco41_R_AMALT3"><caption>R_AMALT3 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_AMALT3</strong><br/>(irreversble)</td><td><strong>Amylomaltase (maltopentaose)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Degradation of polysaccharides<br/><strong>locus_tag: </strong>b3416<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3546008..3548092)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3403; JW3379; malA<br/><strong>EC_number: </strong>2.4.1.25<br/><strong>product: </strong>4-alpha-glucanotransferase (amylomaltase)<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000272 - polysaccharide catabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>malQ<br/><strong>protein_id: </strong>NP_417875.1<br/>" style="color: green;">b3416</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1013)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_maltpt_c + M_malt_c > M_glc__D_c + M_malthx_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Maltopentaose + Maltose C12H22O11 > D-Glucose + Maltohexaose </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_maltpt_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltopentaose</strong></td></tr><tr><td>charge: 0</td><td>C30H52O26</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malt_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glc__D_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_malthx_c</strong></td><td bgcolor="#FFCCCC"><strong>Maltohexaose</strong></td></tr><tr><td>charge: 0</td><td>C36H62O31</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/AMALT3">http://identifiers.org/bigg.reaction/AMALT3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN-14261">http://identifiers.org/biocyc/META:RXN-14261</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.4.1.25">http://identifiers.org/ec-code/2.4.1.25</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95802">http://identifiers.org/metanetx.reaction/MNXR95802</a></td></tr></table><hr/><table id="bigg2-eco41_R_AMALT4"><caption>R_AMALT4 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_AMALT4</strong><br/>(irreversble)</td><td><strong>Amylomaltase (maltohexaose)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Degradation of polysaccharides<br/><strong>locus_tag: </strong>b3416<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3546008..3548092)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3403; JW3379; malA<br/><strong>EC_number: </strong>2.4.1.25<br/><strong>product: </strong>4-alpha-glucanotransferase (amylomaltase)<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000272 - polysaccharide catabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>malQ<br/><strong>protein_id: </strong>NP_417875.1<br/>" style="color: green;">b3416</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1013)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_malt_c + M_malthx_c > M_glc__D_c + M_malthp_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Maltose C12H22O11 + Maltohexaose > D-Glucose + Maltoheptaose </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malt_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malthx_c</strong></td><td bgcolor="#CCFFCC"><strong>Maltohexaose</strong></td></tr><tr><td>charge: 0</td><td>C36H62O31</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glc__D_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_malthp_c</strong></td><td bgcolor="#FFCCCC"><strong>Maltoheptaose</strong></td></tr><tr><td>charge: 0</td><td>C42H72O36</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/AMALT4">http://identifiers.org/bigg.reaction/AMALT4</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:MALTODEG-RXN">http://identifiers.org/biocyc/META:MALTODEG-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.4.1.25">http://identifiers.org/ec-code/2.4.1.25</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95803">http://identifiers.org/metanetx.reaction/MNXR95803</a></td></tr></table><hr/><table id="bigg2-eco41_R_ASPabc"><caption>R_ASPabc (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ASPabc</strong><br/>(irreversble)</td><td><strong>L-aspartate transport via ABC system</strong></td></tr><tr><td><strong>Association</strong></td><td>((b0654 and b0655 and b0652 and <span title="<strong>function: </strong>transport; Transport of small molecules: Amino acids, amines<br/><strong>locus_tag: </strong>b0653<br/><strong>GO_component: </strong>GO:0019866 - organelle inner membrane; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>complement(684478..685152)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0646; JW0648<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>glutamate/aspartate transport system permease; membrane component of ABC superfamily<br/><strong>GO_process: </strong>GO:0006532 - aspartate biosynthetic process; GO:0006537 - glutamate biosynthetic process<br/><strong>gene: </strong>gltK<br/><strong>product: </strong>glutamate and aspartate transporter subunit<br/><strong>protein_id: </strong>NP_415186.1<br/>" style="color: green;">b0653</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1164)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_asp__L_e + M_h2o_c + M_atp_c > M_pi_c + M_asp__L_c + M_adp_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Aspartate + H2O H2O + ATP C10H12N5O13P3 > Phosphate + L-Aspartate + ADP C10H12N5O10P2 + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_asp__L_e</strong></td><td bgcolor="#CCFFCC"><strong>L-Aspartate</strong></td></tr><tr><td>charge: -1</td><td>C4H6NO4</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_asp__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Aspartate</strong></td></tr><tr><td>charge: -1</td><td>C4H6NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_adp_c</strong></td><td bgcolor="#FFCCCC"><strong>ADP C10H12N5O10P2</strong></td></tr><tr><td>charge: -3</td><td>C10H12N5O10P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ASPabc">http://identifiers.org/bigg.reaction/ASPabc</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:TRANS-RXN0-222">http://identifiers.org/biocyc/META:TRANS-RXN0-222</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/3.6.3.21">http://identifiers.org/ec-code/3.6.3.21</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR96103">http://identifiers.org/metanetx.reaction/MNXR96103</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29039#1">http://identifiers.org/rhea/29039#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29040#1">http://identifiers.org/rhea/29040#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29041#1">http://identifiers.org/rhea/29041#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29042#1">http://identifiers.org/rhea/29042#1</a></td></tr></table><hr/><table id="bigg2-eco41_R_CYSS"><caption>R_CYSS (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_CYSS</strong><br/>(irreversble)</td><td><strong>Cysteine synthase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Cysteine<br/><strong>locus_tag: </strong>b2414<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>2530431..2531402<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>cysZ; ECK2409; JW2407<br/><strong>EC_number: </strong>2.5.1.47<br/><strong>note: </strong>cysteine synthase A, O-acetylserine sulfhydrolase A<br/><strong>GO_process: </strong>GO:0019344 - cysteine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>cysK<br/><strong>product: </strong>cysteine synthase A, O-acetylserine sulfhydrolase A subunit<br/><strong>protein_id: </strong>NP_416909.1<br/>" style="color: green;">b2414</span> or b2421))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0846)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_h2s_c + M_acser_c > M_ac_c + M_cys__L_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Hydrogen sulfide + O-Acetyl-L-serine > Acetate + L-Cysteine + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2s_c</strong></td><td bgcolor="#CCFFCC"><strong>Hydrogen sulfide</strong></td></tr><tr><td>charge: 0</td><td>H2S</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_acser_c</strong></td><td bgcolor="#CCFFCC"><strong>O-Acetyl-L-serine</strong></td></tr><tr><td>charge: 0</td><td>C5H9NO4</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ac_c</strong></td><td bgcolor="#FFCCCC"><strong>Acetate</strong></td></tr><tr><td>charge: -1</td><td>C2H3O2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_cys__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Cysteine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO2S</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/CYSS">http://identifiers.org/bigg.reaction/CYSS</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:ACSERLY-RXN">http://identifiers.org/biocyc/META:ACSERLY-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.5.1.47">http://identifiers.org/ec-code/2.5.1.47</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.5.1.65">http://identifiers.org/ec-code/2.5.1.65</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00897">http://identifiers.org/kegg.reaction/R00897</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR97007">http://identifiers.org/metanetx.reaction/MNXR97007</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14829">http://identifiers.org/rhea/14829</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14830">http://identifiers.org/rhea/14830</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14831">http://identifiers.org/rhea/14831</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14832">http://identifiers.org/rhea/14832</a></td></tr></table><hr/><table id="bigg2-eco41_R_DHAPT"><caption>R_DHAPT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_DHAPT</strong><br/>(irreversble)</td><td><strong>Dihydroxyacetone phosphotransferase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1200 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1198 and b1199))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_dha_c > M_pyr_c + M_dhap_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + Dihydroxyacetone > Pyruvate + Dihydroxyacetone phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_dha_c</strong></td><td bgcolor="#CCFFCC"><strong>Dihydroxyacetone</strong></td></tr><tr><td>charge: 0</td><td>C3H6O3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_dhap_c</strong></td><td bgcolor="#FFCCCC"><strong>Dihydroxyacetone phosphate</strong></td></tr><tr><td>charge: -2</td><td>C3H5O6P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/DHAPT">http://identifiers.org/bigg.reaction/DHAPT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:2.7.1.121-RXN">http://identifiers.org/biocyc/META:2.7.1.121-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.1.121">http://identifiers.org/ec-code/2.7.1.121</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01012">http://identifiers.org/kegg.reaction/R01012</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR97364">http://identifiers.org/metanetx.reaction/MNXR97364</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18381">http://identifiers.org/rhea/18381</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18382">http://identifiers.org/rhea/18382</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18383">http://identifiers.org/rhea/18383</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18384">http://identifiers.org/rhea/18384</a></td></tr></table><hr/><table id="bigg2-eco41_R_DHBSr"><caption>R_DHBSr (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_DHBSr</strong><br/>(reversible)</td><td><strong>2,3-dihydroxybenzoate adenylate synthase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Biosynthesis of cofactors, carriers: Enterochelin<br/><strong>locus_tag: </strong>b0594<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>625293..626903<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0587; JW0586<br/><strong>EC_number: </strong>2.7.7.58<br/><strong>note: </strong>2,3-dihydroxybenzoate-AMP ligase<br/><strong>GO_process: </strong>GO:0009239 - enterobactin biosynthetic process; GO:0019184 - nonribosomal peptide biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>entE<br/><strong>product: </strong>2,3-dihydroxybenzoate-AMP ligase component of enterobactin synthase multienzyme complex<br/><strong>protein_id: </strong>NP_415126.1<br/>" style="color: green;">b0594</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0761)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_atp_c + M_23dhb_c = M_ppi_c + M_23dhba_c </td></tr><tr><td><strong>Equation Name</strong></td><td>ATP C10H12N5O13P3 + 2,3-Dihydroxybenzoate = Diphosphate + (2,3-Dihydroxybenzoyl)adenylate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_23dhb_c</strong></td><td bgcolor="#CCFFCC"><strong>2,3-Dihydroxybenzoate</strong></td></tr><tr><td>charge: -1</td><td>C7H5O4</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ppi_c</strong></td><td bgcolor="#FFCCCC"><strong>Diphosphate</strong></td></tr><tr><td>charge: -3</td><td>HO7P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_23dhba_c</strong></td><td bgcolor="#FFCCCC"><strong>(2,3-Dihydroxybenzoyl)adenylate</strong></td></tr><tr><td>charge: -2</td><td>C17H16N5O10P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/DHBSr">http://identifiers.org/bigg.reaction/DHBSr</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:DHBAMPLIG-RXN">http://identifiers.org/biocyc/META:DHBAMPLIG-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.7.58">http://identifiers.org/ec-code/2.7.7.58</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01504">http://identifiers.org/kegg.reaction/R01504</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR97369">http://identifiers.org/metanetx.reaction/MNXR97369</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20229">http://identifiers.org/rhea/20229</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20230">http://identifiers.org/rhea/20230</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20231">http://identifiers.org/rhea/20231</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20232">http://identifiers.org/rhea/20232</a></td></tr></table><hr/><table id="bigg2-eco41_R_FRUpts"><caption>R_FRUpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_FRUpts</strong><br/>(irreversble)</td><td><strong>D-fructose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2169 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b2167))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_fru_e + M_pep_c > M_f1p_c + M_pyr_c </td></tr><tr><td><strong>Equation Name</strong></td><td>D-Fructose + Phosphoenolpyruvate > D-Fructose 1-phosphate + Pyruvate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_fru_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Fructose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_f1p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Fructose 1-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H11O9P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/FRUpts">http://identifiers.org/bigg.reaction/FRUpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR99661">http://identifiers.org/metanetx.reaction/MNXR99661</a></td></tr></table><hr/><table id="bigg2-eco41_R_FRUpts2"><caption>R_FRUpts2 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_FRUpts2</strong><br/>(irreversble)</td><td><strong>Fructose transport via PEP:Pyr PTS (f6p generating)</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1817 and b1819 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1818))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_fru_e + M_pep_c > M_pyr_c + M_f6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>D-Fructose + Phosphoenolpyruvate > Pyruvate + D-Fructose 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_fru_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Fructose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_f6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Fructose 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H11O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/FRUpts2">http://identifiers.org/bigg.reaction/FRUpts2</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR99662">http://identifiers.org/metanetx.reaction/MNXR99662</a></td></tr></table><hr/><table id="bigg2-eco41_R_GALTpts"><caption>R_GALTpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GALTpts</strong><br/>(irreversble)</td><td><strong>Galactitol transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2092 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b2094 and b2093))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_galt_e > M_pyr_c + M_galt1p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + Galactitol > Pyruvate + Galactitol 1-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_galt_e</strong></td><td bgcolor="#CCFFCC"><strong>Galactitol</strong></td></tr><tr><td>charge: 0</td><td>C6H14O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_galt1p_c</strong></td><td bgcolor="#FFCCCC"><strong>Galactitol 1-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H13O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GALTpts">http://identifiers.org/bigg.reaction/GALTpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100018">http://identifiers.org/metanetx.reaction/MNXR100018</a></td></tr></table><hr/><table id="bigg2-eco41_R_GAMpts"><caption>R_GAMpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GAMpts</strong><br/>(irreversble)</td><td><strong>D-glucosamine transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1819 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1817 and b1818))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_gam_e > M_pyr_c + M_gam6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + D-Glucosamine > Pyruvate + D-Glucosamine 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_gam_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Glucosamine</strong></td></tr><tr><td>charge: 1</td><td>C6H14NO5</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_gam6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucosamine 6-phosphate</strong></td></tr><tr><td>charge: -1</td><td>C6H13NO8P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GAMpts">http://identifiers.org/bigg.reaction/GAMpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100034">http://identifiers.org/metanetx.reaction/MNXR100034</a></td></tr></table><hr/><table id="bigg2-eco41_R_GLCP"><caption>R_GLCP (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GLCP</strong><br/>(irreversble)</td><td><strong>Glycogen phosphorylase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Macromolecule synthesis, modification: Polysaccharides - (cytoplasmic)<br/><strong>locus_tag: </strong>b3428<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3562157..3564604)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3414; glgY; JW3391<br/><strong>EC_number: </strong>2.4.1.1<br/><strong>product: </strong>glycogen phosphorylase<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000271 - polysaccharide biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>glgP<br/><strong>protein_id: </strong>NP_417886.1<br/>" style="color: green;">b3428</span> or <span title="<strong>function: </strong>enzyme; Degradation of small molecules: Carbon compounds<br/><strong>locus_tag: </strong>b3417<br/><strong>raw_location: </strong>complement(3548102..3550495)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>blu; ECK3404; JW5689; malA<br/><strong>EC_number: </strong>2.4.1.1<br/><strong>product: </strong>maltodextrin phosphorylase<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0000272 - polysaccharide catabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>malP<br/><strong>protein_id: </strong>YP_026218.1<br/>" style="color: red;">b3417</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1012)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_glycogen_c + M_pi_c > M_g1p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Glycogen C6H10O5 + Phosphate > D-Glucose 1-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glycogen_c</strong></td><td bgcolor="#CCFFCC"><strong>Glycogen C6H10O5</strong></td></tr><tr><td>charge: 0</td><td>C6H10O5</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pi_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_g1p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose 1-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H11O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GLCP">http://identifiers.org/bigg.reaction/GLCP</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100214">http://identifiers.org/metanetx.reaction/MNXR100214</a></td></tr></table><hr/><table id="bigg2-eco41_R_GLCpts"><caption>R_GLCpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GLCpts</strong><br/>(irreversble)</td><td><strong>D-glucose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>(((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1101) or (<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1819 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1817 and b1818) or (<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1621)))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_glc__D_e > M_g6p_c + M_pyr_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + D-Glucose > D-Glucose 6-phosphate + Pyruvate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glc__D_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Glucose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_g6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Glucose 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H11O9P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GLCpts">http://identifiers.org/bigg.reaction/GLCpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100237">http://identifiers.org/metanetx.reaction/MNXR100237</a></td></tr></table><hr/><table id="bigg2-eco41_R_GLNabc"><caption>R_GLNabc (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GLNabc</strong><br/>(irreversble)</td><td><strong>L-glutamine transport via ABC system</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>transport; Transport of small molecules: Amino acids, amines<br/><strong>locus_tag: </strong>b0810<br/><strong>GO_component: </strong>GO:0019866 - organelle inner membrane; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>complement(845683..846342)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0799; JW0795<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>glutamine high-affinity transport system; membrane component<br/><strong>GO_process: </strong>GO:0006542 - glutamine biosynthetic process<br/><strong>gene: </strong>glnP<br/><strong>product: </strong>glutamine transporter subunit<br/><strong>protein_id: </strong>NP_415331.1<br/>" style="color: red;">b0810</span> and <span title="<strong>function: </strong>transport; Transport of small molecules: Amino acids, amines<br/><strong>locus_tag: </strong>b0811<br/><strong>GO_component: </strong>GO:0042597 - periplasmic space<br/><strong>raw_location: </strong>complement(846481..847227)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0800; JW0796<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>periplasmic glutamine-binding protein; permease<br/><strong>GO_process: </strong>GO:0006542 - glutamine biosynthetic process<br/><strong>gene: </strong>glnH<br/><strong>product: </strong>glutamine transporter subunit<br/><strong>protein_id: </strong>NP_415332.1<br/>" style="color: green;">b0811</span> and <span title="<strong>function: </strong>transport; Transport of small molecules: Amino acids, amines<br/><strong>locus_tag: </strong>b0809<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(844964..845686)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0798; JW0794<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>ATP-binding component of glutamine high-affinity transport system<br/><strong>GO_process: </strong>GO:0006542 - glutamine biosynthetic process<br/><strong>gene: </strong>glnQ<br/><strong>product: </strong>glutamine transporter subunit<br/><strong>protein_id: </strong>NP_415330.1<br/>" style="color: green;">b0809</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1492 and stu0606)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_gln__L_e + M_h2o_c + M_atp_c > M_pi_c + M_gln__L_c + M_adp_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Glutamine + H2O H2O + ATP C10H12N5O13P3 > Phosphate + L-Glutamine + ADP C10H12N5O10P2 + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_gln__L_e</strong></td><td bgcolor="#CCFFCC"><strong>L-Glutamine</strong></td></tr><tr><td>charge: 0</td><td>C5H10N2O3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_gln__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamine</strong></td></tr><tr><td>charge: 0</td><td>C5H10N2O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_adp_c</strong></td><td bgcolor="#FFCCCC"><strong>ADP C10H12N5O10P2</strong></td></tr><tr><td>charge: -3</td><td>C10H12N5O10P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GLNabc">http://identifiers.org/bigg.reaction/GLNabc</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:ABC-12-RXN">http://identifiers.org/biocyc/META:ABC-12-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/3.6.3.21">http://identifiers.org/ec-code/3.6.3.21</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100258">http://identifiers.org/metanetx.reaction/MNXR100258</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29895#1">http://identifiers.org/rhea/29895#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29896#1">http://identifiers.org/rhea/29896#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29897#1">http://identifiers.org/rhea/29897#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29898#1">http://identifiers.org/rhea/29898#1</a></td></tr></table><hr/><table id="bigg2-eco41_R_GLU5K"><caption>R_GLU5K (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GLU5K</strong><br/>(irreversble)</td><td><strong>Glutamate 5-kinase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Proline<br/><strong>locus_tag: </strong>b0242<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>259612..260715<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0243; JW0232; pro<br/><strong>EC_number: </strong>2.7.2.11<br/><strong>product: </strong>gamma-glutamate kinase<br/><strong>GO_process: </strong>GO:0006561 - proline biosynthetic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 6089111<br/><strong>protein_id: </strong>NP_414777.1<br/><strong>gene: </strong>proB<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b0242</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1711)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_glu__L_c + M_atp_c > M_adp_c + M_glu5p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Glutamate + ATP C10H12N5O13P3 > ADP C10H12N5O10P2 + L-Glutamate 5-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glu__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_adp_c</strong></td><td bgcolor="#FFCCCC"><strong>ADP C10H12N5O10P2</strong></td></tr><tr><td>charge: -3</td><td>C10H12N5O10P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glu5p_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamate 5-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C5H8NO7P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GLU5K">http://identifiers.org/bigg.reaction/GLU5K</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:GLUTKIN-RXN">http://identifiers.org/biocyc/META:GLUTKIN-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.2.11">http://identifiers.org/ec-code/2.7.2.11</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00239">http://identifiers.org/kegg.reaction/R00239</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100277">http://identifiers.org/metanetx.reaction/MNXR100277</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14877">http://identifiers.org/rhea/14877</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14878">http://identifiers.org/rhea/14878</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14879">http://identifiers.org/rhea/14879</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14880">http://identifiers.org/rhea/14880</a></td></tr></table><hr/><table id="bigg2-eco41_R_GLUabc"><caption>R_GLUabc (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_GLUabc</strong><br/>(irreversble)</td><td><strong>L-glutamate transport via ABC system</strong></td></tr><tr><td><strong>Association</strong></td><td>((b0654 and b0655 and b0652 and <span title="<strong>function: </strong>transport; Transport of small molecules: Amino acids, amines<br/><strong>locus_tag: </strong>b0653<br/><strong>GO_component: </strong>GO:0019866 - organelle inner membrane; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>complement(684478..685152)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0646; JW0648<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>glutamate/aspartate transport system permease; membrane component of ABC superfamily<br/><strong>GO_process: </strong>GO:0006532 - aspartate biosynthetic process; GO:0006537 - glutamate biosynthetic process<br/><strong>gene: </strong>gltK<br/><strong>product: </strong>glutamate and aspartate transporter subunit<br/><strong>protein_id: </strong>NP_415186.1<br/>" style="color: green;">b0653</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1164)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_glu__L_e + M_h2o_c + M_atp_c > M_glu__L_c + M_pi_c + M_adp_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Glutamate + H2O H2O + ATP C10H12N5O13P3 > L-Glutamate + Phosphate + ADP C10H12N5O10P2 + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glu__L_e</strong></td><td bgcolor="#CCFFCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glu__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_adp_c</strong></td><td bgcolor="#FFCCCC"><strong>ADP C10H12N5O10P2</strong></td></tr><tr><td>charge: -3</td><td>C10H12N5O10P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/GLUabc">http://identifiers.org/bigg.reaction/GLUabc</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:ABC-13-RXN">http://identifiers.org/biocyc/META:ABC-13-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/3.6.3.21">http://identifiers.org/ec-code/3.6.3.21</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100299">http://identifiers.org/metanetx.reaction/MNXR100299</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29035#1">http://identifiers.org/rhea/29035#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29036#1">http://identifiers.org/rhea/29036#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29037#1">http://identifiers.org/rhea/29037#1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/29038#1">http://identifiers.org/rhea/29038#1</a></td></tr></table><hr/><table id="bigg2-eco41_R_ICDHyr"><caption>R_ICDHyr (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ICDHyr</strong><br/>(reversible)</td><td><strong>Isocitrate dehydrogenase (NADP)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: TCA cycle; Phage or Prophage Related<br/><strong>locus_tag: </strong>b1136<br/><strong>raw_location: </strong>1194346..1195596<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK1122; icdA; icdE; JW1122<br/><strong>EC_number: </strong>1.1.1.42<br/><strong>note: </strong>isocitrate dehydrogenase, specific for NADP+<br/><strong>GO_process: </strong>GO:0009061 - anaerobic respiration; GO:0006099 - tricarboxylic acid cycle<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_415654.1<br/><strong>gene: </strong>icd<br/><strong>product: </strong>e14 prophage; isocitrate dehydrogenase, specific for NADP+<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b1136</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1266)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_icit_c + M_nadp_c = M_nadph_c + M_co2_c + M_akg_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Isocitrate + Nicotinamide adenine dinucleotide phosphate = Nicotinamide adenine dinucleotide phosphate - reduced + CO2 CO2 + 2-Oxoglutarate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_icit_c</strong></td><td bgcolor="#CCFFCC"><strong>Isocitrate</strong></td></tr><tr><td>charge: -3</td><td>C6H5O7</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_nadp_c</strong></td><td bgcolor="#CCFFCC"><strong>Nicotinamide adenine dinucleotide phosphate</strong></td></tr><tr><td>charge: -3</td><td>C21H25N7O17P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_nadph_c</strong></td><td bgcolor="#FFCCCC"><strong>Nicotinamide adenine dinucleotide phosphate - reduced</strong></td></tr><tr><td>charge: -4</td><td>C21H26N7O17P3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_co2_c</strong></td><td bgcolor="#FFCCCC"><strong>CO2 CO2</strong></td></tr><tr><td>charge: 0</td><td>CO2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_akg_c</strong></td><td bgcolor="#FFCCCC"><strong>2-Oxoglutarate</strong></td></tr><tr><td>charge: -2</td><td>C5H4O5</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ICDHyr">http://identifiers.org/bigg.reaction/ICDHyr</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/1.1.1.42">http://identifiers.org/ec-code/1.1.1.42</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00267">http://identifiers.org/kegg.reaction/R00267</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100781">http://identifiers.org/metanetx.reaction/MNXR100781</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19629">http://identifiers.org/rhea/19629</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19630">http://identifiers.org/rhea/19630</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19631">http://identifiers.org/rhea/19631</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19632">http://identifiers.org/rhea/19632</a></td></tr></table><hr/><table id="bigg2-eco41_R_ILETA"><caption>R_ILETA (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_ILETA</strong><br/>(reversible)</td><td><strong>Isoleucine transaminase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Isoleucine, Valine<br/><strong>locus_tag: </strong>b3770<br/><strong>raw_location: </strong>3950507..3951436<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3762; JW5606<br/><strong>EC_number: </strong>2.6.1.42<br/><strong>product: </strong>branched-chain amino-acid aminotransferase<br/><strong>GO_process: </strong>GO:0006523 - alanine biosynthetic process; GO:0009098 - leucine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>ilvE<br/><strong>protein_id: </strong>YP_026247.1<br/>" style="color: green;">b3770</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0590)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_akg_c + M_ile__L_c = M_glu__L_c + M_3mop_c </td></tr><tr><td><strong>Equation Name</strong></td><td>2-Oxoglutarate + L-Isoleucine = L-Glutamate + (S)-3-Methyl-2-oxopentanoate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_akg_c</strong></td><td bgcolor="#CCFFCC"><strong>2-Oxoglutarate</strong></td></tr><tr><td>charge: -2</td><td>C5H4O5</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ile__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Isoleucine</strong></td></tr><tr><td>charge: 0</td><td>C6H13NO2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glu__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_3mop_c</strong></td><td bgcolor="#FFCCCC"><strong>(S)-3-Methyl-2-oxopentanoate</strong></td></tr><tr><td>charge: -1</td><td>C6H9O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/ILETA">http://identifiers.org/bigg.reaction/ILETA</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERILEU-RXN">http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERILEU-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.42">http://identifiers.org/ec-code/2.6.1.42</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R02199">http://identifiers.org/kegg.reaction/R02199</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR96228">http://identifiers.org/metanetx.reaction/MNXR96228</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24801">http://identifiers.org/rhea/24801</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24802">http://identifiers.org/rhea/24802</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24803">http://identifiers.org/rhea/24803</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24804">http://identifiers.org/rhea/24804</a></td></tr></table><hr/><table id="bigg2-eco41_R_Kt2r"><caption>R_Kt2r (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_Kt2r</strong><br/>(reversible)</td><td><strong>Potassium reversible transport  via proton symport</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>transport; Transport of small molecules: Cations; Phage or Prophage Related<br/><strong>locus_tag: </strong>b1363<br/><strong>GO_component: </strong>GO:0019866 - organelle inner membrane; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>1421806..1423263<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK1362; JW1358<br/><strong>transl_table: </strong>11<br/><strong>product: </strong>Rac prophage; potassium transporter subunit<br/><strong>gene: </strong>trkG<br/><strong>protein_id: </strong>NP_415881.1<br/>" style="color: red;">b1363</span> or <span title="<strong>function: </strong>transport; Transport of small molecules: Cations<br/><strong>locus_tag: </strong>b3849<br/><strong>GO_component: </strong>GO:0019866 - organelle inner membrane; GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>4031168..4032619<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3841; JW5576; sapJ<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>potassium uptake, requires TrkE<br/><strong>gene: </strong>trkH<br/><strong>product: </strong>potassium transporter<br/><strong>protein_id: </strong>YP_026273.1<br/>" style="color: green;">b3849</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0264)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_k_e + M_h_e = M_h_c + M_k_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Potassium + H+ = H+ + Potassium </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_k_e</strong></td><td bgcolor="#CCFFCC"><strong>Potassium</strong></td></tr><tr><td>charge: 1</td><td>K</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h_e</strong></td><td bgcolor="#CCFFCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_k_c</strong></td><td bgcolor="#FFCCCC"><strong>Potassium</strong></td></tr><tr><td>charge: 1</td><td>K</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/Kt2r">http://identifiers.org/bigg.reaction/Kt2r</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:TRANS-RXN-3">http://identifiers.org/biocyc/META:TRANS-RXN-3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR100951">http://identifiers.org/metanetx.reaction/MNXR100951</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28490">http://identifiers.org/rhea/28490</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28491">http://identifiers.org/rhea/28491</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28492">http://identifiers.org/rhea/28492</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28493">http://identifiers.org/rhea/28493</a></td></tr></table><hr/><table id="bigg2-eco41_R_LCADi_copy1"><caption>R_LCADi_copy1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_LCADi_copy1</strong><br/>(reversible)</td><td><strong></strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: Fermentation<br/><strong>locus_tag: </strong>b0356<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(377686..378795)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>adhC; ECK0353; JW0347<br/><strong>EC_number: </strong>1.1.1.1 1.1.1.284<br/><strong>note: </strong>alcohol dehydrogenase class III; formaldehyde dehydrogenase, glutathione-dependent<br/><strong>GO_process: </strong>GO:0006113 - fermentation<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 1731906<br/><strong>protein_id: </strong>NP_414890.1<br/><strong>gene: </strong>frmA<br/><strong>product: </strong>alcohol dehydrogenase class III/glutathione-dependent formaldehyde dehydrogenase<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b0356</span> or b1415 or b1300 or b3588))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0936)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_h2o_c + M_lald__L_c + M_nad_c = M_lac__L_c + (2.0) M_h_c + M_nadh_c </td></tr><tr><td><strong>Equation Name</strong></td><td>H2O H2O + L-Lactaldehyde + Nicotinamide adenine dinucleotide = L-Lactate + (2.0) H+ + Nicotinamide adenine dinucleotide - reduced </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_lald__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Lactaldehyde</strong></td></tr><tr><td>charge: 0</td><td>C3H6O2</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_nad_c</strong></td><td bgcolor="#CCFFCC"><strong>Nicotinamide adenine dinucleotide</strong></td></tr><tr><td>charge: -1</td><td>C21H26N7O14P2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_lac__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Lactate</strong></td></tr><tr><td>charge: -1</td><td>C3H5O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_nadh_c</strong></td><td bgcolor="#FFCCCC"><strong>Nicotinamide adenine dinucleotide - reduced</strong></td></tr><tr><td>charge: -2</td><td>C21H27N7O14P2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:LACTALDDEHYDROG-RXN">http://identifiers.org/biocyc/META:LACTALDDEHYDROG-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/1.2.1.22">http://identifiers.org/ec-code/1.2.1.22</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01446">http://identifiers.org/kegg.reaction/R01446</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101018">http://identifiers.org/metanetx.reaction/MNXR101018</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14277">http://identifiers.org/rhea/14277</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14278">http://identifiers.org/rhea/14278</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14279">http://identifiers.org/rhea/14279</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14280">http://identifiers.org/rhea/14280</a></td></tr></table><hr/><table id="bigg2-eco41_R_LEUTAi"><caption>R_LEUTAi (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_LEUTAi</strong><br/>(irreversble)</td><td><strong>Leucine transaminase (irreversible)</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Isoleucine, Valine<br/><strong>locus_tag: </strong>b3770<br/><strong>raw_location: </strong>3950507..3951436<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3762; JW5606<br/><strong>EC_number: </strong>2.6.1.42<br/><strong>product: </strong>branched-chain amino-acid aminotransferase<br/><strong>GO_process: </strong>GO:0006523 - alanine biosynthetic process; GO:0009098 - leucine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>ilvE<br/><strong>protein_id: </strong>YP_026247.1<br/>" style="color: green;">b3770</span> or b4054))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0590)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_glu__L_c + M_4mop_c > M_leu__L_c + M_akg_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Glutamate + 4-Methyl-2-oxopentanoate > L-Leucine + 2-Oxoglutarate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_glu__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_4mop_c</strong></td><td bgcolor="#CCFFCC"><strong>4-Methyl-2-oxopentanoate</strong></td></tr><tr><td>charge: -1</td><td>C6H9O3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_leu__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Leucine</strong></td></tr><tr><td>charge: 0</td><td>C6H13NO2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_akg_c</strong></td><td bgcolor="#FFCCCC"><strong>2-Oxoglutarate</strong></td></tr><tr><td>charge: -2</td><td>C5H4O5</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/LEUTAi">http://identifiers.org/bigg.reaction/LEUTAi</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERLEU-RXN">http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERLEU-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.42">http://identifiers.org/ec-code/2.6.1.42</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.6">http://identifiers.org/ec-code/2.6.1.6</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.67">http://identifiers.org/ec-code/2.6.1.67</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01090">http://identifiers.org/kegg.reaction/R01090</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR96229">http://identifiers.org/metanetx.reaction/MNXR96229</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18321">http://identifiers.org/rhea/18321</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18322">http://identifiers.org/rhea/18322</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18323">http://identifiers.org/rhea/18323</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18324">http://identifiers.org/rhea/18324</a></td></tr></table><hr/><table id="bigg2-eco41_R_MALTpts"><caption>R_MALTpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_MALTpts</strong><br/>(irreversble)</td><td><strong>Maltose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1621 and b4036))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_malt_e + M_pep_c > M_malt6p_c + M_pyr_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Maltose C12H22O11 + Phosphoenolpyruvate > Maltose 6'-phosphate + Pyruvate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_malt_e</strong></td><td bgcolor="#CCFFCC"><strong>Maltose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_malt6p_c</strong></td><td bgcolor="#FFCCCC"><strong>Maltose 6'-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C12H21O14P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/MALTpts">http://identifiers.org/bigg.reaction/MALTpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101363">http://identifiers.org/metanetx.reaction/MNXR101363</a></td></tr></table><hr/><table id="bigg2-eco41_R_MANpts"><caption>R_MANpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_MANpts</strong><br/>(irreversble)</td><td><strong>D-mannose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b1817 and b1819 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b1818))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_man_e > M_pyr_c + M_man6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + D-Mannose > Pyruvate + D-Mannose 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_man_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Mannose</strong></td></tr><tr><td>charge: 0</td><td>C6H12O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_man6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Mannose 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H11O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/MANpts">http://identifiers.org/bigg.reaction/MANpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101401">http://identifiers.org/metanetx.reaction/MNXR101401</a></td></tr></table><hr/><table id="bigg2-eco41_R_MCITS"><caption>R_MCITS (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_MCITS</strong><br/>(irreversble)</td><td><strong>2-methylcitrate synthase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>putative enzyme; Not classified<br/><strong>locus_tag: </strong>b0333<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>349236..350405<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0330; JW0324; yahS; yzzD<br/><strong>EC_number: </strong>2.3.3.1<br/><strong>note: </strong>putative citrate synthase; propionate metabolism?<br/><strong>GO_process: </strong>GO:0019395 - fatty acid oxidation<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>prpC<br/><strong>product: </strong>2-methylcitrate synthase<br/><strong>protein_id: </strong>NP_414867.1<br/>" style="color: green;">b0333</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1267)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_ppcoa_c + M_h2o_c + M_oaa_c > M_2mcit_c + M_h_c + M_coa_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Propanoyl-CoA + H2O H2O + Oxaloacetate > 2-Methylcitrate + H+ + Coenzyme A </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ppcoa_c</strong></td><td bgcolor="#CCFFCC"><strong>Propanoyl-CoA</strong></td></tr><tr><td>charge: -4</td><td>C24H36N7O17P3S</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_oaa_c</strong></td><td bgcolor="#CCFFCC"><strong>Oxaloacetate</strong></td></tr><tr><td>charge: -2</td><td>C4H2O5</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_2mcit_c</strong></td><td bgcolor="#FFCCCC"><strong>2-Methylcitrate</strong></td></tr><tr><td>charge: -3</td><td>C7H7O7</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_coa_c</strong></td><td bgcolor="#FFCCCC"><strong>Coenzyme A</strong></td></tr><tr><td>charge: -4</td><td>C21H32N7O16P3S</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/MCITS">http://identifiers.org/bigg.reaction/MCITS</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:2-METHYLCITRATE-SYNTHASE-RXN">http://identifiers.org/biocyc/META:2-METHYLCITRATE-SYNTHASE-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.3.3.5">http://identifiers.org/ec-code/2.3.3.5</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00931">http://identifiers.org/kegg.reaction/R00931</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101417">http://identifiers.org/metanetx.reaction/MNXR101417</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/23780">http://identifiers.org/rhea/23780</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/23781">http://identifiers.org/rhea/23781</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/23782">http://identifiers.org/rhea/23782</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/23783">http://identifiers.org/rhea/23783</a></td></tr></table><hr/><table id="bigg2-eco41_R_METS_1"><caption>R_METS_1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_METS_1</strong><br/>(irreversble)</td><td><strong>Methionine synthase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Methionine<br/><strong>locus_tag: </strong>b3829<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>4011076..4013337<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3823; JW3805; metB12<br/><strong>EC_number: </strong>2.1.1.14<br/><strong>note: </strong>tetrahydropteroyltriglutamate methyltransferase<br/><strong>GO_process: </strong>GO:0009086 - methionine biosynthetic process; GO:0006556 - S-adenosylmethionine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>metE<br/><strong>product: </strong>5-methyltetrahydropteroyltriglutamate- homocysteine S-methyltransferase<br/><strong>protein_id: </strong>NP_418273.1<br/>" style="color: green;">b3829</span> or b4019))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0785)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_hcys__L_c + M_5mthf_c > M_met__L_c + M_thf_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Homocysteine + 5-Methyltetrahydrofolate > L-Methionine + 5,6,7,8-Tetrahydrofolate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_hcys__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Homocysteine</strong></td></tr><tr><td>charge: 0</td><td>C4H9NO2S</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_5mthf_c</strong></td><td bgcolor="#CCFFCC"><strong>5-Methyltetrahydrofolate</strong></td></tr><tr><td>charge: -1</td><td>C20H24N7O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_met__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Methionine</strong></td></tr><tr><td>charge: 0</td><td>C5H11NO2S</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_thf_c</strong></td><td bgcolor="#FFCCCC"><strong>5,6,7,8-Tetrahydrofolate</strong></td></tr><tr><td>charge: -2</td><td>C19H21N7O6</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/METS_1">http://identifiers.org/bigg.reaction/METS_1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.1.1.13">http://identifiers.org/ec-code/2.1.1.13</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00946">http://identifiers.org/kegg.reaction/R00946</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101481">http://identifiers.org/metanetx.reaction/MNXR101481</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11172">http://identifiers.org/rhea/11172</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11173">http://identifiers.org/rhea/11173</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11174">http://identifiers.org/rhea/11174</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11175">http://identifiers.org/rhea/11175</a></td></tr></table><hr/><table id="bigg2-eco41_R_MICITDr"><caption>R_MICITDr (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_MICITDr</strong><br/>(irreversble)</td><td><strong>2-methylisocitrate dehydratase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: TCA cycle<br/><strong>locus_tag: </strong>b1276<br/><strong>raw_location: </strong>1333855..1336530<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>acn; ECK1271; JW1268<br/><strong>EC_number: </strong>4.2.1.3<br/><strong>note: </strong>aconitate hydrase 1<br/><strong>GO_process: </strong>GO:0009061 - anaerobic respiration; GO:0006099 - tricarboxylic acid cycle<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>acnA<br/><strong>product: </strong>aconitate hydratase 1<br/><strong>protein_id: </strong>NP_415792.1<br/>" style="color: green;">b1276</span> or b0118))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1268)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_2mcacn_c + M_h2o_c > M_micit_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Cis-2-Methylaconitate + H2O H2O > Methylisocitrate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_2mcacn_c</strong></td><td bgcolor="#CCFFCC"><strong>Cis-2-Methylaconitate</strong></td></tr><tr><td>charge: -3</td><td>C7H5O6</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h2o_c</strong></td><td bgcolor="#CCFFCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_micit_c</strong></td><td bgcolor="#FFCCCC"><strong>Methylisocitrate</strong></td></tr><tr><td>charge: -3</td><td>C7H7O7</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/MICITDr">http://identifiers.org/bigg.reaction/MICITDr</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:4.2.1.99-RXN">http://identifiers.org/biocyc/META:4.2.1.99-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.99">http://identifiers.org/ec-code/4.2.1.99</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R04425">http://identifiers.org/kegg.reaction/R04425</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101580">http://identifiers.org/metanetx.reaction/MNXR101580</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/17941">http://identifiers.org/rhea/17941</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/17942">http://identifiers.org/rhea/17942</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/17943">http://identifiers.org/rhea/17943</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/17944">http://identifiers.org/rhea/17944</a></td></tr></table><hr/><table id="bigg2-eco41_R_MNLpts"><caption>R_MNLpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_MNLpts</strong><br/>(irreversble)</td><td><strong>Mannitol transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b3599))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_mnl_e > M_pyr_c + M_mnl1p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + D-Mannitol > Pyruvate + D-Mannitol 1-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_mnl_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Mannitol</strong></td></tr><tr><td>charge: 0</td><td>C6H14O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_mnl1p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Mannitol 1-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H13O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/MNLpts">http://identifiers.org/bigg.reaction/MNLpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR101677">http://identifiers.org/metanetx.reaction/MNXR101677</a></td></tr></table><hr/><table id="bigg2-eco41_R_OBTFL"><caption>R_OBTFL (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_OBTFL</strong><br/>(irreversble)</td><td><strong>2-Oxobutanoate formate lyase</strong></td></tr><tr><td><strong>Association</strong></td><td>(((<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: Anaerobic respiration<br/><strong>locus_tag: </strong>b0902<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(949563..950303)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>act; ECK0893; JW0885<br/><strong>EC_number: </strong>1.97.1.4<br/><strong>product: </strong>pyruvate formate lyase activating enzyme 1<br/><strong>GO_process: </strong>GO:0009061 - anaerobic respiration; GO:0006464 - protein modification process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>pflA<br/><strong>protein_id: </strong>NP_415422.1<br/>" style="color: green;">b0902</span> and b0903) or b3114))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1640)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_2obut_c + M_coa_c > M_ppcoa_c + M_for_c </td></tr><tr><td><strong>Equation Name</strong></td><td>2-Oxobutanoate + Coenzyme A > Propanoyl-CoA + Formate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_2obut_c</strong></td><td bgcolor="#CCFFCC"><strong>2-Oxobutanoate</strong></td></tr><tr><td>charge: -1</td><td>C4H5O3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_coa_c</strong></td><td bgcolor="#CCFFCC"><strong>Coenzyme A</strong></td></tr><tr><td>charge: -4</td><td>C21H32N7O16P3S</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ppcoa_c</strong></td><td bgcolor="#FFCCCC"><strong>Propanoyl-CoA</strong></td></tr><tr><td>charge: -4</td><td>C24H36N7O17P3S</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_for_c</strong></td><td bgcolor="#FFCCCC"><strong>Formate</strong></td></tr><tr><td>charge: -1</td><td>CH1O2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/OBTFL">http://identifiers.org/bigg.reaction/OBTFL</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:KETOBUTFORMLY-RXN">http://identifiers.org/biocyc/META:KETOBUTFORMLY-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.3.1.-">http://identifiers.org/ec-code/2.3.1.-</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.3.1.54">http://identifiers.org/ec-code/2.3.1.54</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R06987">http://identifiers.org/kegg.reaction/R06987</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR102135">http://identifiers.org/metanetx.reaction/MNXR102135</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28054">http://identifiers.org/rhea/28054</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28055">http://identifiers.org/rhea/28055</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28056">http://identifiers.org/rhea/28056</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/28057">http://identifiers.org/rhea/28057</a></td></tr></table><hr/><table id="bigg2-eco41_R_OCBT"><caption>R_OCBT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_OCBT</strong><br/>(reversible)</td><td><strong>Ornithine carbamoyltransferase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Arginine<br/><strong>locus_tag: </strong>b4254<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(4475330..4476334)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK4247; JW4211<br/><strong>EC_number: </strong>2.1.3.3<br/><strong>product: </strong>ornithine carbamoyltransferase 1<br/><strong>GO_process: </strong>GO:0006526 - arginine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>argI<br/><strong>protein_id: </strong>NP_418675.1<br/>" style="color: green;">b4254</span> or <span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Arginine; Phage or Prophage Related<br/><strong>locus_tag: </strong>b0273<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(288525..289529)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>Arg5; argD; ECK0274; JW0266<br/><strong>EC_number: </strong>2.1.3.3<br/><strong>note: </strong>ornithine carbamoyltransferase 2, chain F<br/><strong>GO_process: </strong>GO:0006526 - arginine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>argF<br/><strong>product: </strong>ornithine carbamoyltransferase 2, chain F; CP4-6 prophage<br/><strong>protein_id: </strong>NP_414807.1<br/>" style="color: red;">b0273</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0603)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_cbp_c + M_orn_c = M_citr__L_c + M_h_c + M_pi_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Carbamoyl phosphate + Ornithine = L-Citrulline + H+ + Phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_cbp_c</strong></td><td bgcolor="#CCFFCC"><strong>Carbamoyl phosphate</strong></td></tr><tr><td>charge: -2</td><td>CH2NO5P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_orn_c</strong></td><td bgcolor="#CCFFCC"><strong>Ornithine</strong></td></tr><tr><td>charge: 1</td><td>C5H13N2O2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_citr__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Citrulline</strong></td></tr><tr><td>charge: 0</td><td>C6H13N3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/OCBT">http://identifiers.org/bigg.reaction/OCBT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:ORNCARBAMTRANSFER-RXN">http://identifiers.org/biocyc/META:ORNCARBAMTRANSFER-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN-13482">http://identifiers.org/biocyc/META:RXN-13482</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.1.3.3">http://identifiers.org/ec-code/2.1.3.3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01398">http://identifiers.org/kegg.reaction/R01398</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR102137">http://identifiers.org/metanetx.reaction/MNXR102137</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19513">http://identifiers.org/rhea/19513</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19514">http://identifiers.org/rhea/19514</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19515">http://identifiers.org/rhea/19515</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19516">http://identifiers.org/rhea/19516</a></td></tr></table><hr/><table id="bigg2-eco41_R_PFL"><caption>R_PFL (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_PFL</strong><br/>(irreversble)</td><td><strong>Pyruvate formate lyase</strong></td></tr><tr><td><strong>Association</strong></td><td>((b3114 or (b3951 and b3952) or (<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: Anaerobic respiration<br/><strong>locus_tag: </strong>b0902<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(949563..950303)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>act; ECK0893; JW0885<br/><strong>EC_number: </strong>1.97.1.4<br/><strong>product: </strong>pyruvate formate lyase activating enzyme 1<br/><strong>GO_process: </strong>GO:0009061 - anaerobic respiration; GO:0006464 - protein modification process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>pflA<br/><strong>protein_id: </strong>NP_415422.1<br/>" style="color: green;">b0902</span> and b0903)))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1640)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pyr_c + M_coa_c > M_accoa_c + M_for_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Pyruvate + Coenzyme A > Acetyl-CoA + Formate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pyr_c</strong></td><td bgcolor="#CCFFCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_coa_c</strong></td><td bgcolor="#CCFFCC"><strong>Coenzyme A</strong></td></tr><tr><td>charge: -4</td><td>C21H32N7O16P3S</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_accoa_c</strong></td><td bgcolor="#FFCCCC"><strong>Acetyl-CoA</strong></td></tr><tr><td>charge: -4</td><td>C23H34N7O17P3S</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_for_c</strong></td><td bgcolor="#FFCCCC"><strong>Formate</strong></td></tr><tr><td>charge: -1</td><td>CH1O2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/PFL">http://identifiers.org/bigg.reaction/PFL</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:PYRUVFORMLY-RXN">http://identifiers.org/biocyc/META:PYRUVFORMLY-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.3.1.54">http://identifiers.org/ec-code/2.3.1.54</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00212">http://identifiers.org/kegg.reaction/R00212</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR102514">http://identifiers.org/metanetx.reaction/MNXR102514</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11844">http://identifiers.org/rhea/11844</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11845">http://identifiers.org/rhea/11845</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11846">http://identifiers.org/rhea/11846</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/11847">http://identifiers.org/rhea/11847</a></td></tr></table><hr/><table id="bigg2-eco41_R_PGM"><caption>R_PGM (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_PGM</strong><br/>(reversible)</td><td><strong>Phosphoglycerate mutase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Energy metabolism, carbon: Glycolysis<br/><strong>locus_tag: </strong>b4395<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>4631820..4632467<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK4387; JW4358; ytjC<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>phosphoglyceromutase 2<br/><strong>GO_process: </strong>GO:0006094 - gluconeogenesis; GO:0006096 - glycolysis<br/><strong>gene: </strong>ytjC<br/><strong>product: </strong>predicted phosphoglyceromutase 2, co-factor independent<br/><strong>protein_id: </strong>NP_418812.1<br/>" style="color: green;">b4395</span> or b3612 or b0755))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0697)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_2pg_c = M_3pg_c </td></tr><tr><td><strong>Equation Name</strong></td><td>D-Glycerate 2-phosphate = 3-Phospho-D-glycerate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_2pg_c</strong></td><td bgcolor="#CCFFCC"><strong>D-Glycerate 2-phosphate</strong></td></tr><tr><td>charge: -3</td><td>C3H4O7P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_3pg_c</strong></td><td bgcolor="#FFCCCC"><strong>3-Phospho-D-glycerate</strong></td></tr><tr><td>charge: -3</td><td>C3H4O7P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/PGM">http://identifiers.org/bigg.reaction/PGM</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:3PGAREARR-RXN">http://identifiers.org/biocyc/META:3PGAREARR-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN-15513">http://identifiers.org/biocyc/META:RXN-15513</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/5.4.2.1">http://identifiers.org/ec-code/5.4.2.1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/5.4.2.11">http://identifiers.org/ec-code/5.4.2.11</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/5.4.2.12">http://identifiers.org/ec-code/5.4.2.12</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01518">http://identifiers.org/kegg.reaction/R01518</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR102547">http://identifiers.org/metanetx.reaction/MNXR102547</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/15901">http://identifiers.org/rhea/15901</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/15902">http://identifiers.org/rhea/15902</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/15903">http://identifiers.org/rhea/15903</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/15904">http://identifiers.org/rhea/15904</a></td></tr></table><hr/><table id="bigg2-eco41_R_PHETA1"><caption>R_PHETA1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_PHETA1</strong><br/>(reversible)</td><td><strong>Phenylalanine transaminase</strong></td></tr><tr><td><strong>Association</strong></td><td>((b0928 or <span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Isoleucine, Valine<br/><strong>locus_tag: </strong>b3770<br/><strong>raw_location: </strong>3950507..3951436<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3762; JW5606<br/><strong>EC_number: </strong>2.6.1.42<br/><strong>product: </strong>branched-chain amino-acid aminotransferase<br/><strong>GO_process: </strong>GO:0006523 - alanine biosynthetic process; GO:0009098 - leucine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>ilvE<br/><strong>protein_id: </strong>YP_026247.1<br/>" style="color: green;">b3770</span> or b4054))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0590)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_akg_c + M_phe__L_c = M_glu__L_c + M_phpyr_c </td></tr><tr><td><strong>Equation Name</strong></td><td>2-Oxoglutarate + L-Phenylalanine = L-Glutamate + Phenylpyruvate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_akg_c</strong></td><td bgcolor="#CCFFCC"><strong>2-Oxoglutarate</strong></td></tr><tr><td>charge: -2</td><td>C5H4O5</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_phe__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Phenylalanine</strong></td></tr><tr><td>charge: 0</td><td>C9H11NO2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glu__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_phpyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Phenylpyruvate</strong></td></tr><tr><td>charge: -1</td><td>C9H7O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/PHETA1">http://identifiers.org/bigg.reaction/PHETA1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:PHEAMINOTRANS-RXN">http://identifiers.org/biocyc/META:PHEAMINOTRANS-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN-10814">http://identifiers.org/biocyc/META:RXN-10814</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.1">http://identifiers.org/ec-code/2.6.1.1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.42">http://identifiers.org/ec-code/2.6.1.42</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.5">http://identifiers.org/ec-code/2.6.1.5</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.57">http://identifiers.org/ec-code/2.6.1.57</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.58">http://identifiers.org/ec-code/2.6.1.58</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.9">http://identifiers.org/ec-code/2.6.1.9</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00694">http://identifiers.org/kegg.reaction/R00694</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR102631">http://identifiers.org/metanetx.reaction/MNXR102631</a></td></tr></table><hr/><table id="bigg2-eco41_R_PRPPS"><caption>R_PRPPS (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_PRPPS</strong><br/>(reversible)</td><td><strong>Phosphoribosylpyrophosphate synthetase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Purine ribonucleotide biosynthesis<br/><strong>locus_tag: </strong>b1207<br/><strong>raw_location: </strong>complement(1260151..1261098)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>dnaR; ECK1195; JW1198; prsA<br/><strong>EC_number: </strong>2.7.6.1<br/><strong>note: </strong>phosphoribosylpyrophosphate synthetase<br/><strong>GO_process: </strong>GO:0009152 - purine ribonucleotide biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>prs<br/><strong>product: </strong>phosphoribosylpyrophosphate synthase<br/><strong>protein_id: </strong>NP_415725.1<br/>" style="color: green;">b1207</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1460)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_r5p_c + M_atp_c = M_prpp_c + M_h_c + M_amp_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Alpha-D-Ribose 5-phosphate + ATP C10H12N5O13P3 = 5-Phospho-alpha-D-ribose 1-diphosphate + H+ + AMP C10H12N5O7P </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_r5p_c</strong></td><td bgcolor="#CCFFCC"><strong>Alpha-D-Ribose 5-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C5H9O8P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_prpp_c</strong></td><td bgcolor="#FFCCCC"><strong>5-Phospho-alpha-D-ribose 1-diphosphate</strong></td></tr><tr><td>charge: -5</td><td>C5H8O14P3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_amp_c</strong></td><td bgcolor="#FFCCCC"><strong>AMP C10H12N5O7P</strong></td></tr><tr><td>charge: -2</td><td>C10H12N5O7P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/PRPPS">http://identifiers.org/bigg.reaction/PRPPS</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.6.1">http://identifiers.org/ec-code/2.7.6.1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR103215">http://identifiers.org/metanetx.reaction/MNXR103215</a></td></tr></table><hr/><table id="bigg2-eco41_R_PTPATi"><caption>R_PTPATi (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_PTPATi</strong><br/>(irreversble)</td><td><strong>Pantetheine-phosphate adenylyltransferase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>putative enzyme; Not classified<br/><strong>locus_tag: </strong>b3634<br/><strong>raw_location: </strong>3807848..3808327<br/><strong>GO_function: </strong>GO:0042280 - cell surface antigen activity, host-interacting<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3624; JW3609; kdtB; yicA<br/><strong>EC_number: </strong>2.7.7.3<br/><strong>product: </strong>pantetheine-phosphate adenylyltransferase<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 10480925<br/><strong>protein_id: </strong>NP_418091.1<br/><strong>gene: </strong>coaD<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b3634</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1648)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_h_c + M_atp_c + M_pan4p_c > M_ppi_c + M_dpcoa_c </td></tr><tr><td><strong>Equation Name</strong></td><td>H+ + ATP C10H12N5O13P3 + Pantetheine 4'-phosphate > Diphosphate + Dephospho-CoA </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h_c</strong></td><td bgcolor="#CCFFCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pan4p_c</strong></td><td bgcolor="#CCFFCC"><strong>Pantetheine 4'-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C11H21N2O7PS</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ppi_c</strong></td><td bgcolor="#FFCCCC"><strong>Diphosphate</strong></td></tr><tr><td>charge: -3</td><td>HO7P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_dpcoa_c</strong></td><td bgcolor="#FFCCCC"><strong>Dephospho-CoA</strong></td></tr><tr><td>charge: -2</td><td>C21H33N7O13P2S</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/PTPATi">http://identifiers.org/bigg.reaction/PTPATi</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:PANTEPADENYLYLTRAN-RXN">http://identifiers.org/biocyc/META:PANTEPADENYLYLTRAN-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.7.3">http://identifiers.org/ec-code/2.7.7.3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R03035">http://identifiers.org/kegg.reaction/R03035</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR95892">http://identifiers.org/metanetx.reaction/MNXR95892</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19801">http://identifiers.org/rhea/19801</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19802">http://identifiers.org/rhea/19802</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19803">http://identifiers.org/rhea/19803</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/19804">http://identifiers.org/rhea/19804</a></td></tr></table><hr/><table id="bigg2-eco41_R_SBTpts"><caption>R_SBTpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_SBTpts</strong><br/>(irreversble)</td><td><strong>D-sorbitol transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2703 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b2704 and b2702))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_sbt__D_e > M_pyr_c + M_sbt6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + D-Sorbitol > Pyruvate + D-Sorbitol 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_sbt__D_e</strong></td><td bgcolor="#CCFFCC"><strong>D-Sorbitol</strong></td></tr><tr><td>charge: 0</td><td>C6H14O6</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_sbt6p_c</strong></td><td bgcolor="#FFCCCC"><strong>D-Sorbitol 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C6H13O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/SBTpts">http://identifiers.org/bigg.reaction/SBTpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104290">http://identifiers.org/metanetx.reaction/MNXR104290</a></td></tr></table><hr/><table id="bigg2-eco41_R_SERAT"><caption>R_SERAT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_SERAT</strong><br/>(reversible)</td><td><strong>Serine O-acetyltransferase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Cysteine<br/><strong>locus_tag: </strong>b3607<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(3779764..3780585)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3597; JW3582<br/><strong>EC_number: </strong>2.3.1.30<br/><strong>product: </strong>serine acetyltransferase<br/><strong>GO_process: </strong>GO:0019344 - cysteine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>cysE<br/><strong>protein_id: </strong>NP_418064.1<br/>" style="color: green;">b3607</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0848)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_accoa_c + M_ser__L_c = M_acser_c + M_coa_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Acetyl-CoA + L-Serine = O-Acetyl-L-serine + Coenzyme A </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_accoa_c</strong></td><td bgcolor="#CCFFCC"><strong>Acetyl-CoA</strong></td></tr><tr><td>charge: -4</td><td>C23H34N7O17P3S</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ser__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Serine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_acser_c</strong></td><td bgcolor="#FFCCCC"><strong>O-Acetyl-L-serine</strong></td></tr><tr><td>charge: 0</td><td>C5H9NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_coa_c</strong></td><td bgcolor="#FFCCCC"><strong>Coenzyme A</strong></td></tr><tr><td>charge: -4</td><td>C21H32N7O16P3S</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/SERAT">http://identifiers.org/bigg.reaction/SERAT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:SERINE-O-ACETTRAN-RXN">http://identifiers.org/biocyc/META:SERINE-O-ACETTRAN-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.3.1.30">http://identifiers.org/ec-code/2.3.1.30</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00586">http://identifiers.org/kegg.reaction/R00586</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104275">http://identifiers.org/metanetx.reaction/MNXR104275</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24560">http://identifiers.org/rhea/24560</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24561">http://identifiers.org/rhea/24561</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24562">http://identifiers.org/rhea/24562</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24563">http://identifiers.org/rhea/24563</a></td></tr></table><hr/><table id="bigg2-eco41_R_SHSL1"><caption>R_SHSL1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_SHSL1</strong><br/>(irreversble)</td><td><strong>O-succinylhomoserine lyase (L-cysteine)</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Methionine<br/><strong>locus_tag: </strong>b3939<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>4126695..4127855<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3931; JW3910; met-1<br/><strong>EC_number: </strong>2.5.1.48<br/><strong>note: </strong>cystathionine gamma-synthase<br/><strong>GO_process: </strong>GO:0009063 - cellular amino acid catabolic process; GO:0009086 - methionine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>metB<br/><strong>product: </strong>cystathionine gamma-synthase, PLP-dependent<br/><strong>protein_id: </strong>NP_418374.1<br/>" style="color: green;">b3939</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0847)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_suchms_c + M_cys__L_c > M_cyst__L_c + M_succ_c + M_h_c </td></tr><tr><td><strong>Equation Name</strong></td><td>O-Succinyl-L-homoserine + L-Cysteine > L-Cystathionine + Succinate + H+ </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_suchms_c</strong></td><td bgcolor="#CCFFCC"><strong>O-Succinyl-L-homoserine</strong></td></tr><tr><td>charge: -1</td><td>C8H12NO6</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_cys__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Cysteine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO2S</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_cyst__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Cystathionine</strong></td></tr><tr><td>charge: 0</td><td>C7H14N2O4S</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_succ_c</strong></td><td bgcolor="#FFCCCC"><strong>Succinate</strong></td></tr><tr><td>charge: -2</td><td>C4H4O4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/SHSL1">http://identifiers.org/bigg.reaction/SHSL1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:O-SUCCHOMOSERLYASE-RXN">http://identifiers.org/biocyc/META:O-SUCCHOMOSERLYASE-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.5.1.48">http://identifiers.org/ec-code/2.5.1.48</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R03260">http://identifiers.org/kegg.reaction/R03260</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104380">http://identifiers.org/metanetx.reaction/MNXR104380</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20397">http://identifiers.org/rhea/20397</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20398">http://identifiers.org/rhea/20398</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20399">http://identifiers.org/rhea/20399</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20400">http://identifiers.org/rhea/20400</a></td></tr></table><hr/><table id="bigg2-eco41_R_SPODM"><caption>R_SPODM (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_SPODM</strong><br/>(irreversble)</td><td><strong>Superoxide dismutase</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Detoxification<br/><strong>locus_tag: </strong>b3908<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>4098833..4099453<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3901; JW3879<br/><strong>EC_number: </strong>1.15.1.1<br/><strong>note: </strong>superoxide dismutase, manganese<br/><strong>GO_process: </strong>GO:0006805 - xenobiotic metabolic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>sodA<br/><strong>product: </strong>superoxide dismutase, Mn<br/><strong>protein_id: </strong>NP_418344.3<br/>" style="color: green;">b3908</span> or b1646 or b1656))</td></tr><tr><td><strong>Association new</strong></td><td>(stu0720)</td></tr><tr><td><strong>Equation ID</strong></td><td>(-2.0) M_o2s_c + (-2.0) M_h_c > M_o2_c + M_h2o2_c </td></tr><tr><td><strong>Equation Name</strong></td><td>(-2.0) Superoxide anion + (-2.0) H+ > O2 O2 + Hydrogen peroxide </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_o2s_c</strong></td><td bgcolor="#CCFFCC"><strong>Superoxide anion</strong></td></tr><tr><td>charge: -1</td><td>O2</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_h_c</strong></td><td bgcolor="#CCFFCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_o2_c</strong></td><td bgcolor="#FFCCCC"><strong>O2 O2</strong></td></tr><tr><td>charge: 0</td><td>O2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h2o2_c</strong></td><td bgcolor="#FFCCCC"><strong>Hydrogen peroxide</strong></td></tr><tr><td>charge: 0</td><td>H2O2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/SPODM">http://identifiers.org/bigg.reaction/SPODM</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:SUPEROX-DISMUT-RXN">http://identifiers.org/biocyc/META:SUPEROX-DISMUT-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/1.15.1.1">http://identifiers.org/ec-code/1.15.1.1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00275">http://identifiers.org/kegg.reaction/R00275</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104498">http://identifiers.org/metanetx.reaction/MNXR104498</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20696">http://identifiers.org/rhea/20696</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20697">http://identifiers.org/rhea/20697</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20698">http://identifiers.org/rhea/20698</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/20699">http://identifiers.org/rhea/20699</a></td></tr></table><hr/><table id="bigg2-eco41_R_SUCpts"><caption>R_SUCpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_SUCpts</strong><br/>(irreversble)</td><td><strong>Sucrose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b2429))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_sucr_e > M_pyr_c + M_suc6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + Sucrose C12H22O11 > Pyruvate + Sucrose 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_sucr_e</strong></td><td bgcolor="#CCFFCC"><strong>Sucrose C12H22O11</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_suc6p_c</strong></td><td bgcolor="#FFCCCC"><strong>Sucrose 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C12H21O14P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/SUCpts">http://identifiers.org/bigg.reaction/SUCpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104643">http://identifiers.org/metanetx.reaction/MNXR104643</a></td></tr></table><hr/><table id="bigg2-eco41_R_THZPSN"><caption>R_THZPSN (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_THZPSN</strong><br/>(irreversble)</td><td><strong>Thiazole phosphate synthesis</strong></td></tr><tr><td><strong>Association</strong></td><td>((b3990 and b3991 and b3992 and <span title="<strong>function: </strong>essential for the biosynthesis of 2-selenouridine in tRNA required for the biosynthesis of thiamin, NAD, Fe-S clusters, and thionucleosides 2.2.3 information transfer; RNA related; RNA modification<br/><strong>locus_tag: </strong>b2530<br/><strong>raw_location: </strong>complement(2658339..2659553)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK2527; JW2514; nuvC; yfhO; yzzO<br/><strong>EC_number: </strong>2.8.1.7<br/><strong>note: </strong>cysteine desulfurase; cysteine desulfurase (tRNA sulfurtransferase) PLP-dependent<br/><strong>GO_process: </strong>GO:0009451 - RNA modification<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>YP_026169.1<br/><strong>gene: </strong>iscS<br/><strong>product: </strong>cysteine desulfurase (tRNA sulfurtransferase), PLP-dependent<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2530</span> and b4407 and b0423))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1461)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_tyr__L_c + M_dxyl5p_c + M_cys__L_c + M_atp_c > M_co2_c + M_4mpetz_c + M_h_c + M_ala__L_c + M_4hba_c + M_ppi_c + M_h2o_c + M_amp_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Tyrosine + 1-deoxy-D-xylulose 5-phosphate + L-Cysteine + ATP C10H12N5O13P3 > CO2 CO2 + 4-Methyl-5-(2-phosphoethyl)-thiazole + H+ + L-Alanine + 4-Hydroxy-benzyl alcohol + Diphosphate + H2O H2O + AMP C10H12N5O7P </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_tyr__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Tyrosine</strong></td></tr><tr><td>charge: 0</td><td>C9H11NO3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_dxyl5p_c</strong></td><td bgcolor="#CCFFCC"><strong>1-deoxy-D-xylulose 5-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C5H9O7P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_cys__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Cysteine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO2S</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_atp_c</strong></td><td bgcolor="#CCFFCC"><strong>ATP C10H12N5O13P3</strong></td></tr><tr><td>charge: -4</td><td>C10H12N5O13P3</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_co2_c</strong></td><td bgcolor="#FFCCCC"><strong>CO2 CO2</strong></td></tr><tr><td>charge: 0</td><td>CO2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_4mpetz_c</strong></td><td bgcolor="#FFCCCC"><strong>4-Methyl-5-(2-phosphoethyl)-thiazole</strong></td></tr><tr><td>charge: -2</td><td>C6H8NO4PS</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h_c</strong></td><td bgcolor="#FFCCCC"><strong>H+</strong></td></tr><tr><td>charge: 1</td><td>H</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ala__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Alanine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_4hba_c</strong></td><td bgcolor="#FFCCCC"><strong>4-Hydroxy-benzyl alcohol</strong></td></tr><tr><td>charge: 0</td><td>C7H8O2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ppi_c</strong></td><td bgcolor="#FFCCCC"><strong>Diphosphate</strong></td></tr><tr><td>charge: -3</td><td>HO7P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h2o_c</strong></td><td bgcolor="#FFCCCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_amp_c</strong></td><td bgcolor="#FFCCCC"><strong>AMP C10H12N5O7P</strong></td></tr><tr><td>charge: -2</td><td>C10H12N5O7P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/THZPSN">http://identifiers.org/bigg.reaction/THZPSN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104858">http://identifiers.org/metanetx.reaction/MNXR104858</a></td></tr></table><hr/><table id="bigg2-eco41_R_TREpts"><caption>R_TREpts (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_TREpts</strong><br/>(irreversble)</td><td><strong>Trehalose transport via PEP:Pyr PTS</strong></td></tr><tr><td><strong>Association</strong></td><td>((<span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2416<br/><strong>raw_location: </strong>2532088..2533815<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2411; JW2409<br/><strong>EC_number: </strong>2.7.3.9<br/><strong>note: </strong>PEP-protein phosphotransferase system enzyme I<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416911.1<br/><strong>gene: </strong>ptsI<br/><strong>product: </strong>PEP-protein phosphotransferase of PTS system (enzyme I)<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2416</span> and b2417 and <span title="<strong>function: </strong>enzyme; Transport of small molecules: Carbohydrates, organic acids, alcohols<br/><strong>locus_tag: </strong>b2415<br/><strong>raw_location: </strong>2531786..2532043<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ctr; ECK2410; Hpr; iex?; JW2408<br/><strong>transl_table: </strong>11<br/><strong>note: </strong>PTS system protein HPr<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646,9868784,10094700<br/><strong>gene: </strong>ptsH<br/><strong>product: </strong>phosphohistidinoprotein-hexose phosphotransferase component of PTS system (Hpr)<br/><strong>protein_id: </strong>NP_416910.1<br/>" style="color: green;">b2415</span> and b4240))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1264 and stu1265)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_tre_e > M_pyr_c + M_tre6p_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + Trehalose > Pyruvate + Alpha,alpha'-Trehalose 6-phosphate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_tre_e</strong></td><td bgcolor="#CCFFCC"><strong>Trehalose</strong></td></tr><tr><td>charge: 0</td><td>C12H22O11</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pyr_c</strong></td><td bgcolor="#FFCCCC"><strong>Pyruvate</strong></td></tr><tr><td>charge: -1</td><td>C3H3O3</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_tre6p_c</strong></td><td bgcolor="#FFCCCC"><strong>Alpha,alpha'-Trehalose 6-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C12H21O14P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/TREpts">http://identifiers.org/bigg.reaction/TREpts</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.7.1.69">http://identifiers.org/ec-code/2.7.1.69</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104931">http://identifiers.org/metanetx.reaction/MNXR104931</a></td></tr></table><hr/><table id="bigg2-eco41_R_TRPS1"><caption>R_TRPS1 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_TRPS1</strong><br/>(irreversble)</td><td><strong>Tryptophan synthase (indoleglycerol phosphate)</strong></td></tr><tr><td><strong>Association</strong></td><td>((b1260 and <span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Tryptophan<br/><strong>locus_tag: </strong>b1261<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(1315246..1316439)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK1255; JW1253<br/><strong>EC_number: </strong>4.2.1.20<br/><strong>note: </strong>tryptophan synthase, beta protein<br/><strong>GO_process: </strong>GO:0000162 - tryptophan biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>trpB<br/><strong>product: </strong>tryptophan synthase, beta subunit<br/><strong>protein_id: </strong>NP_415777.1<br/>" style="color: green;">b1261</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1588)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_ser__L_c + M_3ig3p_c > M_g3p_c + M_trp__L_c + M_h2o_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Serine + C'-(3-Indolyl)-glycerol 3-phosphate > Glyceraldehyde 3-phosphate + L-Tryptophan + H2O H2O </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ser__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Serine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_3ig3p_c</strong></td><td bgcolor="#CCFFCC"><strong>C'-(3-Indolyl)-glycerol 3-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C11H12NO6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_g3p_c</strong></td><td bgcolor="#FFCCCC"><strong>Glyceraldehyde 3-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C3H5O6P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_trp__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Tryptophan</strong></td></tr><tr><td>charge: 0</td><td>C11H12N2O2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h2o_c</strong></td><td bgcolor="#FFCCCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/TRPS1">http://identifiers.org/bigg.reaction/TRPS1</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:TRYPSYN-RXN">http://identifiers.org/biocyc/META:TRYPSYN-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.20">http://identifiers.org/ec-code/4.2.1.20</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R02722">http://identifiers.org/kegg.reaction/R02722</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104343">http://identifiers.org/metanetx.reaction/MNXR104343</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10532">http://identifiers.org/rhea/10532</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10533">http://identifiers.org/rhea/10533</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10534">http://identifiers.org/rhea/10534</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/10535">http://identifiers.org/rhea/10535</a></td></tr></table><hr/><table id="bigg2-eco41_R_TRPS2"><caption>R_TRPS2 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_TRPS2</strong><br/>(irreversble)</td><td><strong>Tryptophan synthase (indole)</strong></td></tr><tr><td><strong>Association</strong></td><td>((b1260 and <span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Tryptophan<br/><strong>locus_tag: </strong>b1261<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(1315246..1316439)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK1255; JW1253<br/><strong>EC_number: </strong>4.2.1.20<br/><strong>note: </strong>tryptophan synthase, beta protein<br/><strong>GO_process: </strong>GO:0000162 - tryptophan biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>trpB<br/><strong>product: </strong>tryptophan synthase, beta subunit<br/><strong>protein_id: </strong>NP_415777.1<br/>" style="color: green;">b1261</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1588)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_ser__L_c + M_indole_c > M_h2o_c + M_trp__L_c </td></tr><tr><td><strong>Equation Name</strong></td><td>L-Serine + Indole > H2O H2O + L-Tryptophan </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ser__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Serine</strong></td></tr><tr><td>charge: 0</td><td>C3H7NO3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_indole_c</strong></td><td bgcolor="#CCFFCC"><strong>Indole</strong></td></tr><tr><td>charge: 0</td><td>C8H7N</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_h2o_c</strong></td><td bgcolor="#FFCCCC"><strong>H2O H2O</strong></td></tr><tr><td>charge: 0</td><td>H2O</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_trp__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Tryptophan</strong></td></tr><tr><td>charge: 0</td><td>C11H12N2O2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/TRPS2">http://identifiers.org/bigg.reaction/TRPS2</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN0-2382">http://identifiers.org/biocyc/META:RXN0-2382</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.122">http://identifiers.org/ec-code/4.2.1.122</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.20">http://identifiers.org/ec-code/4.2.1.20</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00674">http://identifiers.org/kegg.reaction/R00674</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR104946">http://identifiers.org/metanetx.reaction/MNXR104946</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/26434">http://identifiers.org/rhea/26434</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/26435">http://identifiers.org/rhea/26435</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/26436">http://identifiers.org/rhea/26436</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/26437">http://identifiers.org/rhea/26437</a></td></tr></table><hr/><table id="bigg2-eco41_R_TRPS3"><caption>R_TRPS3 (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_TRPS3</strong><br/>(irreversble)</td><td><strong>Tryptophan synthase (indoleglycerol phosphate)</strong></td></tr><tr><td><strong>Association</strong></td><td>((b1260 and <span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Tryptophan<br/><strong>locus_tag: </strong>b1261<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(1315246..1316439)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK1255; JW1253<br/><strong>EC_number: </strong>4.2.1.20<br/><strong>note: </strong>tryptophan synthase, beta protein<br/><strong>GO_process: </strong>GO:0000162 - tryptophan biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>trpB<br/><strong>product: </strong>tryptophan synthase, beta subunit<br/><strong>protein_id: </strong>NP_415777.1<br/>" style="color: green;">b1261</span>))</td></tr><tr><td><strong>Association new</strong></td><td>(stu1588)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_3ig3p_c > M_g3p_c + M_indole_c </td></tr><tr><td><strong>Equation Name</strong></td><td>C'-(3-Indolyl)-glycerol 3-phosphate > Glyceraldehyde 3-phosphate + Indole </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_3ig3p_c</strong></td><td bgcolor="#CCFFCC"><strong>C'-(3-Indolyl)-glycerol 3-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C11H12NO6P</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_g3p_c</strong></td><td bgcolor="#FFCCCC"><strong>Glyceraldehyde 3-phosphate</strong></td></tr><tr><td>charge: -2</td><td>C3H5O6P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_indole_c</strong></td><td bgcolor="#FFCCCC"><strong>Indole</strong></td></tr><tr><td>charge: 0</td><td>C8H7N</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/TRPS3">http://identifiers.org/bigg.reaction/TRPS3</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:RXN0-2381">http://identifiers.org/biocyc/META:RXN0-2381</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.1.2.8">http://identifiers.org/ec-code/4.1.2.8</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/4.2.1.20">http://identifiers.org/ec-code/4.2.1.20</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R02340">http://identifiers.org/kegg.reaction/R02340</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR99889">http://identifiers.org/metanetx.reaction/MNXR99889</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14081">http://identifiers.org/rhea/14081</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14082">http://identifiers.org/rhea/14082</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14083">http://identifiers.org/rhea/14083</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/14084">http://identifiers.org/rhea/14084</a></td></tr></table><hr/><table id="bigg2-eco41_R_UAGCVT"><caption>R_UAGCVT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_UAGCVT</strong><br/>(irreversble)</td><td><strong>UDP-N-acetylglucosamine 1-carboxyvinyltransferase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Murein sacculus, peptidoglycan<br/><strong>locus_tag: </strong>b3189<br/><strong>GO_component: </strong>GO:0009274 - peptidoglycan-based cell wall<br/><strong>raw_location: </strong>complement(3333257..3334516)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3178; JW3156; mrbA; murZ<br/><strong>EC_number: </strong>2.5.1.7<br/><strong>note: </strong>first step in murein biosynthesis;UDP-N-glucosamine 1-carboxyvinyltransferase<br/><strong>GO_process: </strong>GO:0009252 - peptidoglycan biosynthetic process<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 1512209<br/><strong>protein_id: </strong>NP_417656.1<br/><strong>gene: </strong>murA<br/><strong>product: </strong>UDP-N-acetylglucosamine 1-carboxyvinyltransferase<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b3189</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1167)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_pep_c + M_uacgam_c > M_pi_c + M_uaccg_c </td></tr><tr><td><strong>Equation Name</strong></td><td>Phosphoenolpyruvate + UDP-N-acetyl-D-glucosamine > Phosphate + UDP-N-acetyl-3-O-(1-carboxyvinyl)-D-glucosamine </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_pep_c</strong></td><td bgcolor="#CCFFCC"><strong>Phosphoenolpyruvate</strong></td></tr><tr><td>charge: -3</td><td>C3H2O6P</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_uacgam_c</strong></td><td bgcolor="#CCFFCC"><strong>UDP-N-acetyl-D-glucosamine</strong></td></tr><tr><td>charge: -2</td><td>C17H25N3O17P2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_pi_c</strong></td><td bgcolor="#FFCCCC"><strong>Phosphate</strong></td></tr><tr><td>charge: -2</td><td>HO4P</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_uaccg_c</strong></td><td bgcolor="#FFCCCC"><strong>UDP-N-acetyl-3-O-(1-carboxyvinyl)-D-glucosamine</strong></td></tr><tr><td>charge: -3</td><td>C20H26N3O19P2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/UAGCVT">http://identifiers.org/bigg.reaction/UAGCVT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:UDPNACETYLGLUCOSAMENOLPYRTRANS-RXN">http://identifiers.org/biocyc/META:UDPNACETYLGLUCOSAMENOLPYRTRANS-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.5.1.7">http://identifiers.org/ec-code/2.5.1.7</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00660">http://identifiers.org/kegg.reaction/R00660</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR105032">http://identifiers.org/metanetx.reaction/MNXR105032</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18681">http://identifiers.org/rhea/18681</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18682">http://identifiers.org/rhea/18682</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18683">http://identifiers.org/rhea/18683</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/18684">http://identifiers.org/rhea/18684</a></td></tr></table><hr/><table id="bigg2-eco41_R_UDPG4E"><caption>R_UDPG4E (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_UDPG4E</strong><br/>(reversible)</td><td><strong>UDPglucose 4-epimerase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Degradation of small molecules: Carbon compounds<br/><strong>locus_tag: </strong>b0759<br/><strong>GO_component: </strong>GO:0005737 - cytoplasm<br/><strong>raw_location: </strong>complement(790262..791278)<br/><strong>GO_function: </strong>GO:0042280 - cell surface antigen activity, host-interacting<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK0748; galD; JW0742<br/><strong>EC_number: </strong>5.1.3.2<br/><strong>product: </strong>UDP-galactose-4-epimerase<br/><strong>GO_process: </strong>GO:0016052 - carbohydrate catabolic process; GO:0009242 - colanic acid biosynthetic process; GO:0006012 - galactose metabolic process; GO:0009435 - NAD biosynthetic process; GO:0009269 - response to desiccation<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 4613352,9119048<br/><strong>protein_id: </strong>NP_415280.3<br/><strong>gene: </strong>galE<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b0759</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu1169)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_udpg_c = M_udpgal_c </td></tr><tr><td><strong>Equation Name</strong></td><td>UDPglucose = UDPgalactose </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_udpg_c</strong></td><td bgcolor="#CCFFCC"><strong>UDPglucose</strong></td></tr><tr><td>charge: -2</td><td>C15H22N2O17P2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_udpgal_c</strong></td><td bgcolor="#FFCCCC"><strong>UDPgalactose</strong></td></tr><tr><td>charge: -2</td><td>C15H22N2O17P2</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/UDPG4E">http://identifiers.org/bigg.reaction/UDPG4E</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:UDPGLUCEPIM-RXN">http://identifiers.org/biocyc/META:UDPGLUCEPIM-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/5.1.3.2">http://identifiers.org/ec-code/5.1.3.2</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00291">http://identifiers.org/kegg.reaction/R00291</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR105057">http://identifiers.org/metanetx.reaction/MNXR105057</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22168">http://identifiers.org/rhea/22168</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22169">http://identifiers.org/rhea/22169</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22170">http://identifiers.org/rhea/22170</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/22171">http://identifiers.org/rhea/22171</a></td></tr></table><hr/><table id="bigg2-eco41_R_UPPRT"><caption>R_UPPRT (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_UPPRT</strong><br/>(irreversble)</td><td><strong>Uracil phosphoribosyltransferase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Salvage of nucleosides and nucleotides<br/><strong>locus_tag: </strong>b2498<br/><strong>raw_location: </strong>complement(2618268..2618894)<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK2494; JW2483; uraP<br/><strong>EC_number: </strong>2.4.2.9<br/><strong>product: </strong>uracil phosphoribosyltransferase<br/><strong>GO_process: </strong>GO:0015949 - nucleobase, nucleoside and nucleotide interconversion<br/><strong>experiment: </strong>N-terminus verified by Edman degradation: PMID 9298646<br/><strong>protein_id: </strong>NP_416993.2<br/><strong>gene: </strong>upp<br/><strong>transl_table: </strong>11<br/>" style="color: green;">b2498</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0355)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_prpp_c + M_ura_c > M_ppi_c + M_ump_c </td></tr><tr><td><strong>Equation Name</strong></td><td>5-Phospho-alpha-D-ribose 1-diphosphate + Uracil > Diphosphate + UMP C9H11N2O9P </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_prpp_c</strong></td><td bgcolor="#CCFFCC"><strong>5-Phospho-alpha-D-ribose 1-diphosphate</strong></td></tr><tr><td>charge: -5</td><td>C5H8O14P3</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_ura_c</strong></td><td bgcolor="#CCFFCC"><strong>Uracil</strong></td></tr><tr><td>charge: 0</td><td>C4H4N2O2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ppi_c</strong></td><td bgcolor="#FFCCCC"><strong>Diphosphate</strong></td></tr><tr><td>charge: -3</td><td>HO7P2</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_ump_c</strong></td><td bgcolor="#FFCCCC"><strong>UMP C9H11N2O9P</strong></td></tr><tr><td>charge: -2</td><td>C9H11N2O9P</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/UPPRT">http://identifiers.org/bigg.reaction/UPPRT</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:URACIL-PRIBOSYLTRANS-RXN">http://identifiers.org/biocyc/META:URACIL-PRIBOSYLTRANS-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.4.2.9">http://identifiers.org/ec-code/2.4.2.9</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R00966">http://identifiers.org/kegg.reaction/R00966</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR105142">http://identifiers.org/metanetx.reaction/MNXR105142</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/13017">http://identifiers.org/rhea/13017</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/13018">http://identifiers.org/rhea/13018</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/13019">http://identifiers.org/rhea/13019</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/13020">http://identifiers.org/rhea/13020</a></td></tr></table><hr/><table id="bigg2-eco41_R_VALTA"><caption>R_VALTA (<a href="#page-top">top</a>)</caption><tr><td><strong>Organism</strong></td><td><strong>bigg2-eco41</strong></td></tr><tr><td><strong>R_VALTA</strong><br/>(reversible)</td><td><strong>Valine transaminase</strong></td></tr><tr><td><strong>Association</strong></td><td><span title="<strong>function: </strong>enzyme; Amino acid biosynthesis: Isoleucine, Valine<br/><strong>locus_tag: </strong>b3770<br/><strong>raw_location: </strong>3950507..3951436<br/><strong>codon_start: </strong>1<br/><strong>gene_synonym: </strong>ECK3762; JW5606<br/><strong>EC_number: </strong>2.6.1.42<br/><strong>product: </strong>branched-chain amino-acid aminotransferase<br/><strong>GO_process: </strong>GO:0006523 - alanine biosynthetic process; GO:0009098 - leucine biosynthetic process<br/><strong>transl_table: </strong>11<br/><strong>gene: </strong>ilvE<br/><strong>protein_id: </strong>YP_026247.1<br/>" style="color: green;">b3770</span> </td></tr><tr><td><strong>Association new</strong></td><td>(stu0590)</td></tr><tr><td><strong>Equation ID</strong></td><td>M_akg_c + M_val__L_c = M_glu__L_c + M_3mob_c </td></tr><tr><td><strong>Equation Name</strong></td><td>2-Oxoglutarate + L-Valine = L-Glutamate + 3-Methyl-2-oxobutanoate </td></tr><tr><td colspan="2" align="center"><strong>Substrates</strong></td></tr><tr><td bgcolor="#CCFFCC"><strong>M_akg_c</strong></td><td bgcolor="#CCFFCC"><strong>2-Oxoglutarate</strong></td></tr><tr><td>charge: -2</td><td>C5H4O5</td></tr><tr><td bgcolor="#CCFFCC"><strong>M_val__L_c</strong></td><td bgcolor="#CCFFCC"><strong>L-Valine</strong></td></tr><tr><td>charge: 0</td><td>C5H11NO2</td></tr><tr><td colspan="2" align="center"><strong>Products</strong></td></tr><tr><td bgcolor="#FFCCCC"><strong>M_glu__L_c</strong></td><td bgcolor="#FFCCCC"><strong>L-Glutamate</strong></td></tr><tr><td>charge: -1</td><td>C5H8NO4</td></tr><tr><td bgcolor="#FFCCCC"><strong>M_3mob_c</strong></td><td bgcolor="#FFCCCC"><strong>3-Methyl-2-oxobutanoate</strong></td></tr><tr><td>charge: -1</td><td>C5H7O3</td></tr><tr><td colspan="2" align="center"><strong>Annotations</strong></td></tr><tr><td colspan="2" align="left">RDF references (opens in browser)</td></tr><tr><td>is</td><td><a href="http://identifiers.org/bigg.reaction/VALTA">http://identifiers.org/bigg.reaction/VALTA</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERVAL-RXN">http://identifiers.org/biocyc/META:BRANCHED-CHAINAMINOTRANSFERVAL-RXN</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.42">http://identifiers.org/ec-code/2.6.1.42</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/ec-code/2.6.1.6">http://identifiers.org/ec-code/2.6.1.6</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/kegg.reaction/R01214">http://identifiers.org/kegg.reaction/R01214</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/metanetx.reaction/MNXR96230">http://identifiers.org/metanetx.reaction/MNXR96230</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24813">http://identifiers.org/rhea/24813</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24814">http://identifiers.org/rhea/24814</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24815">http://identifiers.org/rhea/24815</a></td></tr><tr><td>is</td><td><a href="http://identifiers.org/rhea/24816">http://identifiers.org/rhea/24816</a></td></tr></table> <p>&nbsp;</p> <p><small>Generated by MetaDraft 0.9.3 &copy; <a href="https://systemsbioinformatics.github.io/metatoolkit/">MetaToolkit</a>.</small></p></body></html>